--- 1.30/BBS_GHU.bbs	2022-06-29 16:26:16.000000000 +0200
+++ 1.31/BBS_GHU.bbs	2022-06-29 17:54:18.000000000 +0200
@@ -2832,6 +2832,8 @@
   sprite: s32'ghu200_03', 3
   hit: 
   EffectForLanding: 100, 1, 1
+  physicsImpulseX: 80000
+  horizontalMomentum: 60
   sprite: s32'ghu200_04', 2
   sprite: s32'ghu200_05', 2
   physicsImpulseX: 60000
@@ -3188,7 +3190,7 @@
     storeValue: (VARIABLE), 51, (STATIC), 1
     callSubroutine: s32'Potential_LvMAX'
     endUpon: 
-  sprite: s32'ghu230_00', 3
+  sprite: s32'ghu230_00', 2
   Hair: 5
   sprite: s32'ghu230_01', 2
   sprite: s32'ghu230_02', 2
@@ -3329,7 +3331,7 @@
   sprite: s32'ghu203_04', 2
   EffectForLanding: 100, 1, 0
   callSubroutine: s32'cmnRendaChain_Clear_Idling'
-  physicsImpulseX: 40000
+  physicsImpulseX: 55000
   physicsXImpulseIfVelocityMatchesSign: 25000
   hit: 
   sprite: s32'ghu203_05', 2
@@ -3732,8 +3734,8 @@
   sprite: s32'ghu233_00', 1
   Hair: 5
   HairReset: 4
-  sprite: s32'ghu233_01', 2
-  sprite: s32'ghu233_02', 2
+  sprite: s32'ghu233_01', 1
+  sprite: s32'ghu233_02', 1
   beginLabel: s32'atk'
   sprite: s32'ghu233_03', 3
   randomVoiceLine2: s16'vxxx106', 100, s16'vxxx107', 100, s16'', 0, s16'', 0
@@ -3795,8 +3797,8 @@
   hit: 
   sprite: s32'ghu260_04', 2
   sprite: s32'ghu260_04', 2
-  recoveryState: 
   sprite: s32'ghu260_05', 4
+  recoveryState: 
   sprite: s32'ghu260_06', 3
   sprite: s32'ghu260_07', 3
   exitState: 
@@ -3971,14 +3973,14 @@
   randomVoiceLine2: s16'vxxx106', 100, s16'vxxx107', 100, s16'', 0, s16'', 0
   createObject: s32'jumpkidan_start', 0
   physicsImpulseX: -100000
-  physicsImpulseY: 24000
+  physicsImpulseY: 10000
   setGravity: 1800
   horizontalMomentum: 80
   cmnSfx: s32'ARC_BTL_CMN_GK_JumpAir'
   sprite: s32'ghu233_03', 2
   landingRecovery: 5, 1
-  sprite: s32'ghu233_04', 3
-  sprite: s32'ghu233_05', 2
+  sprite: s32'ghu233_04', 1
+  sprite: s32'ghu233_05', 1
   resetGravity: 
   sprite: s32'ghu233_08', 2
   createObject: s32'jumpkidan', 0
@@ -3988,20 +3990,24 @@
   sprite: s32'ghu233_09', 2
   sprite: s32'ghu233_10', 2
   sprite: s32'ghu233_11', 2
-  sprite: s32'ghu233_09', 2
-  sprite: s32'ghu233_10', 2
-  sprite: s32'ghu233_11', 2
-  sprite: s32'ghu233_12', 3
-  sprite: s32'ghu233_13', 3
-  sprite: s32'ghu233_14', 3
-  sprite: s32'ghu022_04', 3
-  sprite: s32'ghu022_05', 3
+  sprite: s32'ghu233_12', 2
+  sprite: s32'ghu233_13', 2
+  sprite: s32'ghu233_14', 2
+  sprite: s32'ghu022_04', 2
+  sprite: s32'ghu022_05', 2
   endState: 
 
 beginState: s32'NmlAtkAir2D'
   upon: (IMMEDIATE)
     AtkIgnoreCaution: 1
+    addWhiffCancelOption: s32'NmlAtk5C'
+    addWhiffCancelOption: s32'NmlAtk2C'
+    addWhiffCancelOption: s32'NmlAtk3C'
     addWhiffCancelOption: s32'NmlAtk2D'
+    addWhiffCancelOption: s32'NmlAtk5D'
+    addWhiffCancelOption: s32'Potential_AB'
+    addWhiffCancelOption: s32'Potential_CD'
+    addWhiffCancelOption: s32'UltimateKamehameha'
     upon: (BEFORE_EXIT)
       StopBankSE: 0, 10
       endUpon: 
@@ -4090,6 +4096,9 @@
     AtkAngle: 40
     AtkNoSousai: 1
     AtkShakeCamera: s32'ShakeTate', 600, 0, 20, 5
+    if: (VARIABLE), 3
+      untechAmount: 33
+      endIf: 
     damageSoundEffect: s32'ARC_BTL_CMN_Hit_XLarge'
     guardSoundEffect: s32'ARC_BTL_CMN_Guard_Large'
     callSubroutine: s32'Potential_LvMAX2'
@@ -4100,7 +4109,9 @@
   Hair: 5
   HairReset: 4
   headInvul: 1
+  spriteAddTime: 2, 2, 3
   sprite: s32'ghu400_12', 3
+  spriteAddTime: 2, 2, 3
   sprite: s32'ghu400_12', 1
   voiceLine: s16'vxxx202'
   characterSfx: s32'ARC_BTL_GHU_JetUpper_A'
@@ -4116,8 +4127,14 @@
   sprite: s32'ghu400_15', 1
   physicsImpulseY: 35000
   setGravity: 2400
+  if: (VARIABLE), 3
+    physicsImpulseY: 42000
+    setGravity: 3600
+    endIf: 
   sprite: s32'ghu400_16', 1
-  landingRecovery: 12, 1
+  ifNot: (VARIABLE), 3
+    landingRecovery: 12, 1
+    endIf: 
   sprite: s32'ghu400_17', 2
   sprite: s32'ghu400_18', 2
   sprite: s32'ghu400_19', 3
@@ -6346,6 +6363,9 @@
   storeValue: (VARIABLE), 202, (STATIC), 1
   clearRegisteredUponCode: (ON_ATTACK_COLLISION_SOUSAI)
   SenkoInputSpecial: 20
+  enableWhiffCancel: 1
+  addWhiffCancelOption: s32'CmnActHomingDash'
+  whiffCancelOptionBufferTime: s32'CmnActHomingDash', 10
   sprite: s32'ghu402_10', 3
   sprite: s32'ghu402_11', 3
   deactivateObjectByName: s32'kamehameha_shadow'
@@ -6564,11 +6584,11 @@
   sprite: s32'ghu430_03', 1
   createObject: s32'ghu_430_eff3', (PLAYER)
   enableSuperArmor: 1
-  superFreeze: 3, 119, 112, 112
+  superFreeze: 3, 119, 140, 140
   upon: (SUPERFREEZE)
     clearRegisteredUponCode: (SUPERFREEZE)
     CameraForceMove: 1
-    cameraMotion: 4660, 330059, 82783, 283, 141, 0, 54, 15, 97, 8, 1
+    cameraMotion: 4660, 330059, 82783, 283, 141, 0, 54, 15, 125, 8, 1
     callSubroutine: s32'cmn1gaugeWLDSTP'
     enableMemberChangeUltimate: 0
     endUpon: 
@@ -6678,10 +6698,8 @@
     endElse: 
   enableMemberChangeUltimate: 1
   sprite: s32'ghu430_14', 8
-  ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), 45, (STATIC), 4
-    superFreeze: 3, 119, 8, 8
-    cameraMotion: 4660, 330059, 82783, 283, 141, 0, 54, 0, 6, 7, 1
-    endIf: 
+  superFreeze: 3, 119, 8, 8
+  cameraMotion: 4660, 330059, 82783, 283, 141, 0, 54, 0, 6, 7, 1
   sprite: s32'ghu430_15', 3
   if: (VARIABLE), 50
     SmartVoiceEx: s16'vxxx213'
