--- 1.30/BBS_SGN.bbs	2022-06-29 16:26:25.000000000 +0200
+++ 1.31/BBS_SGN.bbs	2022-06-29 17:54:21.000000000 +0200
@@ -2139,7 +2139,12 @@
   gotoLabel: s32'downloop'
   beginLabel: s32'landing'
   sprite: s32'sgn010_00', 3
-  jumpToState: s32'Ryuken'
+  if: (VARIABLE), (DmgAtkAngle)
+    jumpToState: s32'SU_Genkidama'
+    endIf: 
+  else: 
+    jumpToState: s32'Ryuken'
+    endIf: 
   endLabel: 
   endState: 
 
@@ -2776,7 +2781,7 @@
   sprite: s32'sgn202_03', 2
   sprite: s32'sgn202_04', 1
   setCarriedMomentumPercentage: 50
-  physicsImpulseX: 30000
+  physicsImpulseX: 60000
   horizontalMomentum: 70
   CommonSetSE: s32'FuriShort', 0, 100
   sprite: s32'sgn202_05', 1
@@ -6393,6 +6398,9 @@
   clearRegisteredUponCode: (BEAM_CLASH)
   triggerUponInObject: s32'SGN_AirKamehameha_Core', (ON_HIT_OR_BLOCK)
   deactivateObjectByName: s32'sgn403_kamehameha_shadow'
+  enableWhiffCancel: 1
+  addWhiffCancelOption: s32'CmnActHomingDash'
+  whiffCancelOptionBufferTime: s32'CmnActHomingDash', 10
   sprite: s32'sgn403_13', 3
   sprite: s32'sgn403_14', 3
   endLabel: 
@@ -8342,23 +8350,23 @@
   ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (BackWallDistance), (STATIC), 650000
     ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (EnemyDistanceX), (STATIC), 650000
       storeValue: (VARIABLE), 46, (STATIC), 1
-      moveToOffset: 32, 100, 2000000, 1000000, 7
+      moveToOffset: 32, 100, 1000000, 1000000, 7
       endIf: 
     else: 
-      moveToOffset: 32, 100, -2000000, 1000000, 7
+      moveToOffset: 32, 100, -1000000, 1000000, 7
       endElse: 
     endIf: 
   else: 
-    moveToOffset: 32, 100, -2000000, 1000000, 7
+    moveToOffset: 32, 100, -1000000, 1000000, 7
     endElse: 
   endSubroutine: 
 
 beginSubroutine: s32'SU_Genkidama_WarpEnd'
   if: (VARIABLE), 46
-    moveToOffset: 32, 100, 2500000, 1000000, 2
+    moveToOffset: 32, 100, 1500000, 1000000, 2
     endIf: 
   else: 
-    moveToOffset: 32, 100, -2500000, 1000000, 2
+    moveToOffset: 32, 100, -1500000, 1000000, 2
     endElse: 
   ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), (PosY), (STATIC), 5000000
     setPositionY: 5000000
@@ -8370,6 +8378,7 @@
 
 beginState: s32'SU_Genkidama'
   upon: (IMMEDIATE)
+    PlayerActionInitializeByCategory: 20, 1
     tensionUsage: 30000
     endUpon: 
   SetMeteoUltimateSkillOffset: 2
