--- 1.30/BBS_GKSEF.bbs	2022-06-29 16:26:18.000000000 +0200
+++ 1.31/BBS_GKSEF.bbs	2022-06-29 17:54:18.000000000 +0200
@@ -251,8 +251,11 @@
       ifNot: (VARIABLE), (Result)
         groundHitEffect: (AIR_FACE_UP)
         airHitEffect: (AIR_FACE_UP)
-        hitAirPushbackX: 20000
+        hitAirPushbackX: 30000
         hitAirPushbackY: 20000
+        wallBounceCount: 1
+        wallBounceXVelocityPercent: 100
+        untechAmount: 24
         endIf: 
       endIf: 
     endUpon: 
@@ -1543,7 +1546,7 @@
     overdriveAttack: 
     callSubroutine: s32'cmnAtkLevel_4_AtkInit'
     callSubroutine: s32'cmnUltimateShot_AtkInit'
-    damage1: 250
+    damage1: 270
     minimalDamagePercent: 31
     hitStop: 2
     groundHitEffect: (AIR_STRONG)
@@ -1826,8 +1829,8 @@
     overdriveAttack: 
     callSubroutine: s32'cmnAtkLevel_4_AtkInit'
     callSubroutine: s32'cmnUltimateShot_AtkInit'
-    damage1: 270
-    minimalDamagePercent: 30
+    damage1: 280
+    minimalDamagePercent: 31
     hitStop: 2
     groundHitEffect: 12
     airHitEffect: 12
@@ -2097,7 +2100,7 @@
     overdriveAttack: 
     callSubroutine: s32'cmnAtkLevel_4_AtkInit'
     callSubroutine: s32'cmnUltimateShot_AtkInit'
-    damage1: 250
+    damage1: 260
     minimalDamagePercent: 30
     hitStop: 2
     groundHitEffect: (AIR_STRONG)
