--- 1.30/BBS_KFS.bbs	2022-06-29 16:26:20.000000000 +0200
+++ 1.31/BBS_KFS.bbs	2022-06-29 17:54:19.000000000 +0200
@@ -2887,6 +2887,10 @@
   sprite: s32'kfs232_04', 2
   physicsImpulseX: 50000
   horizontalMomentum: 80
+  checkInput: (INPUT_ANY_FORWARD)
+  if: (VARIABLE), (Result)
+    physicsImpulseX: 95000
+    endIf: 
   sprite: s32'kfs232_05', 2
   sprite: s32'kfs232_06', 1
   hit: 
@@ -3370,6 +3374,9 @@
   clearRegisteredUponCode: (FRAME_STEP)
   sprite: s32'kfs401_15', 2
   sprite: s32'kfs401_16', 2
+  enableWhiffCancel: 1
+  addWhiffCancelOption: s32'CmnActHomingDash'
+  whiffCancelOptionBufferTime: s32'CmnActHomingDash', 10
   sprite: s32'kfs401_17', 2
   sprite: s32'kfs401_18', 2
   sprite: s32'kfs401_19', 4
@@ -3472,9 +3479,9 @@
   sprite: s32'kfs401_16', 2
   sprite: s32'kfs401_17', 2
   sprite: s32'kfs401_18', 2
-  sprite: s32'kfs401_19', 3
-  sprite: s32'kfs401_20', 3
-  sprite: s32'kfs401_21', 3
+  sprite: s32'kfs401_19', 2
+  sprite: s32'kfs401_20', 2
+  sprite: s32'kfs401_21', 1
   endLabel: 
   exitState: 
   beginLabel: s32'sousai'
@@ -3927,7 +3934,7 @@
   ifOperation: (IS_LESSER), (VARIABLE), (EnemyFrontDistance), (STATIC), 0
     Cell: s32'kfs403_19'
     endIf: 
-  sprite: s32'kfs403_17', 3
+  sprite: s32'kfs403_17', 2
   faceOpponent: 
   ignorePushback: 0
   ScreenAdditionalWidth: 0
@@ -4033,7 +4040,7 @@
   ifOperation: (IS_LESSER), (VARIABLE), (EnemyFrontDistance), (STATIC), 0
     Cell: s32'kfs403_19'
     endIf: 
-  sprite: s32'kfs403_17', 3
+  sprite: s32'kfs403_17', 1
   faceOpponent: 
   ignorePushback: 0
   ScreenAdditionalWidth: 0
