--- 1.30/BBS_GFF.bbs	2022-06-29 16:26:14.000000000 +0200
+++ 1.31/BBS_GFF.bbs	2022-06-29 17:54:17.000000000 +0200
@@ -2899,7 +2899,8 @@
     callSubroutine: s32'cmnNmlAtk5C_AtkInit'
     callSubroutine: s32'cmnNmlAtk5C_Init'
     addGatlingOption: s32'NmlAtk6D'
-    pushbackX: 150
+    pushbackX: 250
+    AtkHitBackForHit: 60
     upon: 49
       CheckSignalName: s32'CameraCombo_5C'
       if: (VARIABLE), (Result)
@@ -2987,6 +2988,11 @@
   sprite: s32'gff232_03', 2
   physicsImpulseX: 80000
   horizontalMomentum: 90
+  checkInput: (INPUT_4)
+  if: (VARIABLE), (Result)
+    pushbackX: 200
+    physicsImpulseX: 10000
+    endIf: 
   EffectForLanding: 100, 1, 0
   sprite: s32'gff232_04', 2
   sprite: s32'gff232_05', 1
@@ -4536,9 +4542,9 @@
     storeValue: (VARIABLE), 45, (STATIC), 10
     callSubroutine: s32'MStrike_Init'
     endUpon: 
-  sprite: s32'gff401_00', 3
+  sprite: s32'gff401_00', 2
   Hair: 5
-  sprite: s32'gff401_01', 3
+  sprite: s32'gff401_01', 2
   sprite: s32'keep', 2
   cmnSfx: s32'ARC_BTL_CMN_GK_Jump'
   sprite: s32'gff401_02', 3
