--- 1.30/BBS_MTNEF.bbs	2022-06-29 16:26:23.000000000 +0200
+++ 1.31/BBS_MTNEF.bbs	2022-06-29 17:54:20.000000000 +0200
@@ -1088,7 +1088,7 @@
     hitAirPushbackY: 15000
     untechAmount: 30
     blockstunAmount: 22
-    wallBounceXVelocityPercent: 0
+    wallBounceXVelocityPercent: 30
     AtkTargetPlayerHitOnly: s32'MTN_Homing_Yoko_Atk', 1
     AtkRange: 50000000, -210800, 0, 0
     AtkSousaiSevereCheck: 1
@@ -1348,8 +1348,8 @@
     groundHitEffect: 12
     airHitEffect: 12
     hitAirPushbackX: 20000
-    hitAirPushbackY: 60000
-    untechAmount: 30
+    hitAirPushbackY: 45000
+    untechAmount: 45
     blockstunAmount: 22
     AtkTargetPlayerHitOnly: s32'MTN_Homing_Ue_Atk', 1
     AtkRange: 50000000, -151400, 0, 0
@@ -1902,9 +1902,10 @@
     airHitEffect: (AIR_STRONG)
     hitAirPushbackX: 90000
     hitAirPushbackY: 15000
-    untechAmount: 30
+    untechAmount: 39
     blockstunAmount: 22
     wallBounceXVelocityPercent: 0
+    AtkWallBoundWaitTime: 14
     AtkTargetPlayerHitOnly: s32'MTN_Kamehameha_Atk', 1
     AtkRange: 50000000, -369899, 0, 0
     AtkSousaiSevereCheck: 1
