--- 1.30/BBS_ASN.bbs	2022-06-29 16:26:08.000000000 +0200
+++ 1.31/BBS_ASN.bbs	2022-06-29 17:54:15.000000000 +0200
@@ -1334,15 +1334,15 @@
   HairReset: 4
   sprite: s32'asn404_01', 1
   sprite: s32'asn404_02', 1
-  sprite: s32'asn404_03', 2
+  sprite: s32'asn404_03', 1
   characterSfx: s32'ARC_BTL_ASN_HellsHeat_SetUp'
-  sprite: s32'asn404_04', 2
-  sprite: s32'asn404_05', 2
-  sprite: s32'asn404_06', 2
+  sprite: s32'asn404_04', 1
+  sprite: s32'asn404_05', 1
+  sprite: s32'asn404_06', 1
   sprite: s32'asn404_07', 1
   sprite: s32'asn404_08', 1
-  sprite: s32'asn404_09', 2
-  sprite: s32'asn404_10', 2
+  sprite: s32'asn404_09', 1
+  sprite: s32'asn404_10', 1
   sprite: s32'asn404_12', 2
   createObject: s32'HHL_core', 0
   AddWindPowerOffset: -1000000, 0, 0, 1000, 0, 20, 30
@@ -2243,13 +2243,13 @@
   sprite: s32'asn200_02', 1
   AttackVoice: 0, -1
   CommonSetSE: s32'FuriShort', 0, 100
-  physicsImpulseX: 5000
+  physicsImpulseX: 25000
   extendPushboxX: 48500
   sprite: s32'asn200_03', 3
   hit: 
   callSubroutine: s32'cmnRendaChain_Clear_Idling'
   sprite: s32'asn200_04', 3
-  horizontalMomentum: 20
+  horizontalMomentum: 40
   recoveryState: 
   sprite: s32'asn200_05', 2
   enableWhiffCancel: 1
@@ -2314,11 +2314,13 @@
   cmnSfx: s32'ARC_BTL_CMN_Furi_Keri'
   AttackVoice: 1, 255
   sprite: s32'asn202_03', 1
+  addVelocityX: 80000
   sprite: s32'asn202_04', 1
   createObject: s32'202Eff', (PLAYER)
   if: (VARIABLE), (IN_BATTLE)
     hit: 
     endIf: 
+  horizontalMomentum: 70
   sprite: s32'asn202_23', 3
   recoveryState: 
   sprite: s32'asn202_24', 4
@@ -2497,11 +2499,11 @@
   hit: 
   callSubroutine: s32'cmnRendaChain_Clear_Idling'
   sprite: s32'asn203_05ex', 2
-  sprite: s32'asn203_06', 4
+  sprite: s32'asn203_06', 3
   horizontalMomentum: 30
   PushCollisionHeight: -1
   recoveryState: 
-  sprite: s32'asn203_07', 4
+  sprite: s32'asn203_07', 3
   sprite: s32'asn203_08', 3
   sprite: s32'asn203_09', 3
   sprite: s32'asn203_10', 3
@@ -2763,6 +2765,7 @@
   sprite: s32'asn232_12', 2
   sprite: s32'asn232_13', 2
   StopBankSE: 0, 20
+  setGravity: 800
   sprite: s32'asn232_14', 2
   sprite: s32'asn232_15', 3
   sprite: s32'asn022_02', 3
@@ -2790,14 +2793,14 @@
   sprite: s32'asn205_00', 1
   Hair: 6
   HairReset: 4
-  sprite: s32'asn205_01', 3
+  sprite: s32'asn205_01', 2
   sprite: s32'asn205_02', 2
   cmnSfx: s32'ARC_BTL_CMN_Charge_Bishi'
-  sprite: s32'asn205_03', 2
+  sprite: s32'asn205_03', 1
   SetCrouch: 1
-  sprite: s32'asn205_04', 2
-  sprite: s32'asn205_05', 2
-  sprite: s32'asn205_06', 2
+  sprite: s32'asn205_04', 1
+  sprite: s32'asn205_05', 1
+  sprite: s32'asn205_06', 1
   voiceLine: s16'vxxx106'
   sprite: s32'asn205_07', 2
   storeValue: (VARIABLE), 46, (STATIC), 0
@@ -3153,7 +3156,10 @@
   physicsImpulseY: 60000
   horizontalMomentum: 90
   verticalMomentum: 90
-  sprite: s32'asn263_02', 4
+  if: (VARIABLE), (EnemyInHitstun)
+    physicsImpulseX: 10000
+    endIf: 
+  sprite: s32'asn263_02', 3
   sprite: s32'asn263_03', 2
   horizontalMomentum: 70
   verticalMomentum: 70
@@ -5026,10 +5032,10 @@
   sprite: s32'asn404_18', 2
   characterSfx: s32'ARC_BTL_ASN_HellsHeat_End'
   sprite: s32'asn404_19', 3
-  sprite: s32'asn404_20', 5
-  sprite: s32'asn404_21', 3
+  sprite: s32'asn404_20', 3
+  sprite: s32'asn404_21', 2
   sprite: s32'asn404_22', 2
-  sprite: s32'asn404_23', 2
+  sprite: s32'asn404_23', 1
   endState: 
 
 beginState: s32'HellsHeat_L'
