--- 1.30/BBS_GNN.bbs	2022-06-29 16:26:18.000000000 +0200
+++ 1.31/BBS_GNN.bbs	2022-06-29 17:54:18.000000000 +0200
@@ -1547,7 +1547,7 @@
   sprite: s32'gnn402_06', 2
   sprite: s32'gnn402_07', 2
   sprite: s32'gnn402_08', 2
-  addTensionUnscaled: 0, 2500
+  addTensionUnscaled: 0, 5000
   sprite: s32'gnn402_09', 5
   recoveryState: 
   sprite: s32'gnn402_10', 5
@@ -1577,7 +1577,7 @@
   sprite: s32'gnn403_06', 2
   sprite: s32'gnn403_07', 2
   sprite: s32'gnn403_08', 2
-  addTensionUnscaled: 0, 2500
+  addTensionUnscaled: 0, 7500
   sprite: s32'gnn403_09', 3
   recoveryState: 
   sprite: s32'gnn403_10', 3
@@ -2260,9 +2260,9 @@
   upon: (IMMEDIATE)
     callSubroutine: s32'cmnNmlAtk5A_AtkInit'
     callSubroutine: s32'cmnNmlAtk5A_Init'
-    removeWhiffCancelOption: s32'NmlAtk5A'
-    removeGatlingOption: s32'NmlAtk5A'
     endUpon: 
+  checkLastStateName: s32'NmlAtk5A'
+  gotoIfTrue: s32'Renda', 2, 0
   sprite: s32'gnn200_00', 1
   sprite: s32'gnn200_01', 1
   sprite: s32'gnn200_02', 2
@@ -2281,6 +2281,24 @@
   sprite: s32'gnn200_08', 2
   enableWhiffCancel: 0
   sprite: s32'gnn200_09', 2
+  exitState: 
+  beginLabel: s32'Renda'
+  sprite: s32'gnn200_08', 2
+  sprite: s32'gnn200_03', 3
+  AttackVoice: 0, -1
+  cmnSfx: s32'ARC_BTL_CMN_Furi_Hook'
+  sprite: s32'gnn200_04', 4
+  createObject: s32'gnn_5AEff', (PLAYER)
+  hit: 
+  callSubroutine: s32'cmnRendaChain_Clear_Idling'
+  sprite: s32'gnn200_05', 3
+  recoveryState: 
+  enableWhiffCancel: 1
+  sprite: s32'gnn200_06', 2
+  sprite: s32'gnn200_07', 2
+  sprite: s32'gnn200_08', 2
+  enableWhiffCancel: 0
+  sprite: s32'gnn200_09', 2
   endState: 
 
 beginState: s32'NmlAtk5A2nd'
@@ -2363,6 +2381,7 @@
     callSubroutine: s32'cmnNmlAtk5A3rd_AtkInit'
     callSubroutine: s32'cmnNmlAtk5A3rd_Init'
     addGatlingOption: s32'NmlAtk5D'
+    pushbackX: -50
     upon: 49
       CheckSignalName: s32'CameraCombo_5A3rd'
       if: (VARIABLE), (Result)
@@ -2586,6 +2605,14 @@
   AttackVoice: 2, -1
   CommonBankSE: 0, s32'ARC_BTL_CMN_ZSpin', 100
   AtkZReversal: 1
+  enableSuperArmor: 1
+  superArmorType: (DODGE)
+  superArmorLoseMid: 0
+  superArmorLoseHigh: 0
+  ExMutekiGuardImpossible: 0
+  superArmorOverdriveGuard: 0
+  superArmorBurstGuard: 0
+  Unknown1869: 0
   upon: (FRAME_STEP)
     if: (VARIABLE), 46
       physicsImpulseX: 40000
@@ -2711,6 +2738,7 @@
   StopBankSE: 0, 10
   sprite: s32'gnn204_12', 4
   AtkZReversal: 0
+  enableSuperArmor: 0
   sprite: s32'gnn204_13', 4
   recoveryState: 
   sprite: s32'gnn204_14', 4
@@ -2840,7 +2868,9 @@
     callSubroutine: s32'tokusentai_Init'
     endUpon: 
   sprite: s32'gnn205_00', 2
-  sprite: s32'gnn205_01', 2
+  sprite: s32'gnn205_01', 1
+  sprite: s32'gnn205_01', 1
+  superArmorKiBlastInvul: 1
   sprite: s32'gnn205_02', 3
   FaceReset: 502
   HairReset: 4
@@ -3063,6 +3093,7 @@
   PushSpeedY: 
   haltMomentum: 1
   sprite: s32'gnn265_01', 3
+  AtkZReversal: 1
   sprite: s32'gnn265_02', 3
   sprite: s32'gnn265_03', 4
   sprite: s32'gnn265_04', 5
@@ -3081,6 +3112,7 @@
   PerSpeedX: 80
   PerSpeedY: 80
   resetGravity: 
+  AtkZReversal: 0
   sprite: s32'gnn265_11', 3
   HairReset: 0
   sprite: s32'gnn265_12', 3
