--- 1.30/BBS_EST.bbs	2022-06-29 16:26:12.000000000 +0200
+++ 1.31/BBS_EST.bbs	2022-06-29 17:54:17.000000000 +0200
@@ -926,15 +926,15 @@
     endIf: 
   if: (VARIABLE), 53
     if: (VARIABLE), (IsSparking)
-      AtkAttackRate: 125, 13, 1
+      AtkAttackRate: 135, 13, 1
       if: (VARIABLE), (CmnCharaReg7)
-        AtkAttackRate: 135, 11, 0
+        AtkAttackRate: 145, 11, 0
         endIf: 
       endIf: 
     else: 
-      AtkAttackRate: 105, 10, 1
+      AtkAttackRate: 115, 10, 1
       if: (VARIABLE), (CmnCharaReg7)
-        AtkAttackRate: 125, 11, 0
+        AtkAttackRate: 135, 11, 0
         endIf: 
       endElse: 
     endIf: 
@@ -1897,7 +1897,12 @@
   gotoLabel: s32'downloop'
   beginLabel: s32'landing'
   sprite: s32'est010_00', 3
-  jumpToState: s32'Gcharge'
+  if: (VARIABLE), (DmgAtkAngle)
+    jumpToState: s32'Gimpact'
+    endIf: 
+  else: 
+    jumpToState: s32'Gcharge'
+    endIf: 
   endLabel: 
   endState: 
 
@@ -3429,7 +3434,12 @@
   sprite: s32'est206_05', 2
   cmnSfx: s32'ARC_BTL_CMN_Furi_SyagamiKeri'
   sprite: s32'est206_06', 3
-  physicsImpulseX: 50000
+  physicsImpulseX: 75000
+  if: (VARIABLE), (EnemyInHitstun)
+    IfBeginCoType: 32, 2, 30
+      PushCollisionHeight: 500000
+      endIf: 
+    endIf: 
   createObject: s32'est206_Smoke_Eff', (PLAYER_LAND)
   sprite: s32'est206_07', 3
   characterSfx: s32'ARC_BTL_EST_Throw_Tukami'
@@ -3441,6 +3451,7 @@
   hit: 
   deactivateObjectByName: s32'est206_Smoke_Eff'
   sprite: s32'est206_28', 3
+  PushCollisionHeight: -1
   recoveryState: 
   sprite: s32'est206_29', 3
   sprite: s32'est206_30', 3
@@ -3657,6 +3668,8 @@
     callSubroutine: s32'cmnNmlAtk5D_Init'
     whiffCancelOptionBufferTime: s32'NmlAtk2D', 10
     addGatlingOption: s32'NmlAtk2D'
+    whiffCancelOptionBufferTime: s32'NmlAtk6C', 10
+    addGatlingOption: s32'NmlAtk6C'
     storeValue: (VARIABLE), 45, (STATIC), 5
     upon: 12
       callSubroutine: s32'cmnKidan_RendaInit'
@@ -3684,9 +3697,9 @@
         endIf: 
       endUpon: 
     endUpon: 
-  sprite: s32'est205_00', 2
-  sprite: s32'est205_01', 2
-  sprite: s32'est205_02', 3
+  sprite: s32'est205_00', 1
+  sprite: s32'est205_01', 1
+  sprite: s32'est205_02', 2
   beginLabel: s32'hasei1'
   sprite: s32'est205_03', 1
   ifNot: (VARIABLE), 48
@@ -3827,6 +3840,7 @@
 beginState: s32'NmlAtk2D'
   upon: (IMMEDIATE)
     callSubroutine: s32'cmnNmlAtk5D_Init'
+    addGatlingOption: s32'NmlAtk6C'
     upon: (FRAME_STEP)
       if: (VARIABLE), 52
         modifyVar: (ADD), (VARIABLE), 52, (STATIC), 1
@@ -3861,6 +3875,7 @@
         SenkoInputSpecial: 20
         whiffCancelOptionBufferTime: s32'NmlAtk5C', 8
         whiffCancelOptionBufferTime: s32'NmlAtk2C', 8
+        whiffCancelOptionBufferTime: s32'NmlAtk6C', 8
         whiffCancelOptionBufferTime: s32'NmlAtk5D', 8
         endIf: 
       endUpon: 
@@ -5378,9 +5393,13 @@
   headInvul: 1
   Hair: 6
   sprite: s32'est403_01', 3
-  sprite: s32'est403_02', 4
-  sprite: s32'est403_03', 4
+  spriteAddTime: 1, 2, 214
+  sprite: s32'est403_02', 3
+  spriteAddTime: 1, 2, 214
+  sprite: s32'est403_03', 3
+  spriteAddTime: 1, 2, 214
   sprite: s32'est403_04', 5
+  spriteAddTime: 1, 2, 214
   voiceLine: s16'vxxx202'
   sprite: s32'est403_05', 2
   HairReset: 4
@@ -7682,7 +7701,9 @@
   physicsImpulseX: 120000
   sprite: s32'est430_34', 2
   sprite: s32'est430_35', 2
-  addWhiffCancelOption: s32'Gimpact_Chain'
+  ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 59, (STATIC), 1
+    addWhiffCancelOption: s32'Gimpact_Chain'
+    endIf: 
   addWhiffCancelOption: s32'Omega'
   beginLabel: s32'6_end'
   sprite: s32'est430_32', 2
@@ -7692,7 +7713,9 @@
   sprite: s32'est430_35', 2
   sprite: s32'est430_36', 2
   horizontalMomentum: 80
-  addWhiffCancelOption: s32'Gimpact_Chain'
+  ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 59, (STATIC), 1
+    addWhiffCancelOption: s32'Gimpact_Chain'
+    endIf: 
   addWhiffCancelOption: s32'Omega'
   ScreenAdditionalWidth: 0
   endLabel: 
@@ -7725,7 +7748,9 @@
   resetGravity: 
   sprite: s32'est430_28', 2
   sprite: s32'est430_29', 2
-  addWhiffCancelOption: s32'Gimpact_Chain'
+  ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 59, (STATIC), 1
+    addWhiffCancelOption: s32'Gimpact_Chain'
+    endIf: 
   addWhiffCancelOption: s32'Omega'
   beginLabel: s32'9_end'
   sprite: s32'est430_26', 2
@@ -7736,7 +7761,9 @@
   sprite: s32'est430_30', 2
   horizontalMomentum: 80
   verticalMomentum: 80
-  addWhiffCancelOption: s32'Gimpact_Chain'
+  ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 59, (STATIC), 1
+    addWhiffCancelOption: s32'Gimpact_Chain'
+    endIf: 
   addWhiffCancelOption: s32'Omega'
   ScreenAdditionalWidth: 0
   endLabel: 
@@ -7773,7 +7800,9 @@
   resetGravity: 
   sprite: s32'est430_40', 2
   sprite: s32'est430_41', 2
-  addWhiffCancelOption: s32'Gimpact_Chain'
+  ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 59, (STATIC), 1
+    addWhiffCancelOption: s32'Gimpact_Chain'
+    endIf: 
   addWhiffCancelOption: s32'Omega'
   upon: (TOUCH_GROUND)
     jumpToLabel: s32'landing'
@@ -7782,14 +7811,18 @@
   sprite: s32'est430_38', 2
   sprite: s32'est430_39', 2
   upon: (TOUCH_GROUND)
-    addWhiffCancelOption: s32'Gimpact_Chain'
+    ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 59, (STATIC), 1
+      addWhiffCancelOption: s32'Gimpact_Chain'
+      endIf: 
     addWhiffCancelOption: s32'Omega'
     jumpToLabel: s32'landing'
     endUpon: 
   sprite: s32'est430_40', 2
   sprite: s32'est430_41', 2
   sprite: s32'est430_38', 2
-  addWhiffCancelOption: s32'Gimpact_Chain'
+  ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 59, (STATIC), 1
+    addWhiffCancelOption: s32'Gimpact_Chain'
+    endIf: 
   addWhiffCancelOption: s32'Omega'
   beginLabel: s32'3_loop'
   sprite: s32'est430_39', 2
@@ -7830,7 +7863,9 @@
   resetGravity: 
   sprite: s32'est430_46', 2
   sprite: s32'est430_47', 2
-  addWhiffCancelOption: s32'Gimpact_Chain'
+  ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 59, (STATIC), 1
+    addWhiffCancelOption: s32'Gimpact_Chain'
+    endIf: 
   addWhiffCancelOption: s32'Omega'
   upon: (TOUCH_GROUND)
     jumpToLabel: s32'landing'
@@ -7839,14 +7874,18 @@
   sprite: s32'est430_44', 2
   sprite: s32'est430_45', 2
   upon: (TOUCH_GROUND)
-    addWhiffCancelOption: s32'Gimpact_Chain'
+    ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 59, (STATIC), 1
+      addWhiffCancelOption: s32'Gimpact_Chain'
+      endIf: 
     addWhiffCancelOption: s32'Omega'
     jumpToLabel: s32'landing'
     endUpon: 
   sprite: s32'est430_46', 2
   sprite: s32'est430_47', 2
   sprite: s32'est430_44', 2
-  addWhiffCancelOption: s32'Gimpact_Chain'
+  ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 59, (STATIC), 1
+    addWhiffCancelOption: s32'Gimpact_Chain'
+    endIf: 
   addWhiffCancelOption: s32'Omega'
   beginLabel: s32'2_loop'
   sprite: s32'est430_45', 2
@@ -7885,7 +7924,9 @@
   resetGravity: 
   sprite: s32'est430_22', 2
   sprite: s32'est430_23', 2
-  addWhiffCancelOption: s32'Gimpact_Chain'
+  ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 59, (STATIC), 1
+    addWhiffCancelOption: s32'Gimpact_Chain'
+    endIf: 
   addWhiffCancelOption: s32'Omega'
   beginLabel: s32'8_end'
   sprite: s32'est430_20', 2
@@ -7894,7 +7935,9 @@
   sprite: s32'est430_23', 2
   sprite: s32'est430_24', 2
   verticalMomentum: 80
-  addWhiffCancelOption: s32'Gimpact_Chain'
+  ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 59, (STATIC), 1
+    addWhiffCancelOption: s32'Gimpact_Chain'
+    endIf: 
   addWhiffCancelOption: s32'Omega'
   endLabel: 
   gotoLabel: s32'end'
@@ -7968,7 +8011,9 @@
   EffectForLandBreak: '0x01000000E80300000000000000000000'
   setNoCollision: 1
   horizontalMomentum: 70
-  addWhiffCancelOption: s32'Gimpact_Chain'
+  ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 59, (STATIC), 1
+    addWhiffCancelOption: s32'Gimpact_Chain'
+    endIf: 
   addWhiffCancelOption: s32'Omega'
   ifOperation: (IS_EQUAL), (VARIABLE), 55, (STATIC), 6
     Cell: s32'est430_36'
