--- 1.30/BBS_GHT.bbs	2022-06-29 16:26:15.000000000 +0200
+++ 1.31/BBS_GHT.bbs	2022-06-29 17:54:17.000000000 +0200
@@ -2794,7 +2794,7 @@
   exPointFReset: (FLOOR_CENTER), 250000, 0
   EffectForDashStart: 130, 1, 0
   createObject: s32'GHT_2BEff', (PLAYER)
-  physicsImpulseX: 60000
+  physicsImpulseX: 80000
   horizontalMomentum: 90
   AttackVoice: 1, -1
   cmnSfx: s32'ARC_BTL_CMN_Furi_SyagamiKeri'
@@ -2879,7 +2879,7 @@
   CommonBankSE: 0, s32'ARC_BTL_CMN_JetDash', 100
   sprite: s32'ght204_05', 2
   createParticle: s32'bg_jump_default', (PLAYER)
-  physicsImpulseX: 40000
+  physicsImpulseX: 60000
   cmnSfx: s32'ARC_BTL_CMN_Furi_YokoKeri'
   StopBankSE: 0, 20
   sprite: s32'ght204_06', 2
@@ -3057,16 +3057,16 @@
       storeValue: (VARIABLE), 45, (STATIC), 0
       endIf: 
     endUpon: 
-  sprite: s32'ght233_00', 2
+  sprite: s32'ght233_00', 1
   Hair: 6
   HairReset: 4
-  sprite: s32'ght233_01', 2
+  sprite: s32'ght233_01', 1
   createObject: s32'crouchkidan_start', 0
-  sprite: s32'ght233_02', 2
+  sprite: s32'ght233_02', 1
   characterSfx: s32'ARC_BTL_GHT_NmtnKame_Chrg'
   landingRecovery: 6, 1
-  sprite: s32'ght233_03', 2
-  sprite: s32'ght233_04', 2
+  sprite: s32'ght233_03', 1
+  sprite: s32'ght233_04', 1
   cmnSfx: s32'ARC_BTL_CMN_GK_Jump'
   sprite: s32'ght233_05', 2
   createParticle: s32'bg_jump_default', (PLAYER)
@@ -5013,7 +5013,7 @@
     airHitEffect: (AIR_LAUNCH)
     hitAirPushbackX: 0
     hitAirPushbackY: 45000
-    untechAmount: 42
+    untechAmount: 44
     AtkHitBackForHit: 10
     pushbackX: 50
     blockstunAmount: 11
@@ -5822,10 +5822,10 @@
     FinishCameraRequest: s32'FinishCamera_NKamehameha'
     tensionUsage: 10000
     endUpon: 
-  sprite: s32'ght430_00', 2
+  sprite: s32'ght430_00', 1
   haltMomentum: 1
-  sprite: s32'ght430_01', 3
-  sprite: s32'ght430_02', 3
+  sprite: s32'ght430_01', 1
+  sprite: s32'ght430_02', 2
   characterSfx: s32'ARC_BTL_GHT_NmtnKame_Basa'
   sprite: s32'ght430_03', 1
   enableSuperArmor: 1
@@ -5968,12 +5968,12 @@
     FinishCameraRequest: s32'FinishCamera_O_Kamehameha'
     tensionUsage: 30000
     endUpon: 
-  sprite: s32'ght431_00', 3
+  sprite: s32'ght431_00', 2
   enableSuperArmor: 1
   haltMomentum: 1
-  sprite: s32'ght431_01', 3
-  sprite: s32'ght431_02', 3
-  sprite: s32'ght431_03', 3
+  sprite: s32'ght431_01', 2
+  sprite: s32'ght431_02', 2
+  sprite: s32'ght431_03', 2
   sprite: s32'ght431_03', 1
   superFreeze: 4, 119, 151, 152
   CellTimeAdd: 151
