--- 1.30/BBS_FRNEF.bbs	2022-06-29 16:26:13.000000000 +0200
+++ 1.31/BBS_FRNEF.bbs	2022-06-29 17:54:17.000000000 +0200
@@ -904,7 +904,7 @@
     callSubroutine: s32'cmnAtkLevel_2_AtkInit'
     callSubroutine: s32'cmn_AirAtkHosei'
     setProjectileDurability: 2
-    damage1: 400
+    damage1: 500
     tensionGainOnConnect: 400
     AtkHoseiTadan: 1
     hitAirPushbackX: 60000
@@ -2643,7 +2643,7 @@
     overdriveAttack: 
     callSubroutine: s32'cmnAtkLevel_4_AtkInit'
     callSubroutine: s32'cmnUltimateShot_AtkInit'
-    damage1: 480
+    damage1: 520
     minimalDamagePercent: 40
     hitStop: 2
     groundHitEffect: 12
