--- 1.30/BBS_BDN.bbs	2022-06-29 16:26:09.000000000 +0200
+++ 1.31/BBS_BDN.bbs	2022-06-29 17:54:16.000000000 +0200
@@ -1474,7 +1474,7 @@
     hitAirPushbackY: 15000
     pushbackX: 100
     hitStop: 12
-    enemyHitstopAddition: 0, 3, 3
+    enemyHitstopAddition: 0, 8, 8
     jumpToStateInObject: 92, s32'atk_end'
     upon: (BEFORE_EXIT)
       ResetWeightDefault: 
@@ -2533,8 +2533,9 @@
   sprite: s32'bdn202_00', 2
   Hair: 6
   HairReset: 4
+  headInvul: 1
   sprite: s32'bdn202_01', 2
-  physicsImpulseX: 80000
+  physicsImpulseX: 120000
   horizontalMomentum: 80
   sprite: s32'bdn202_02', 2
   EffectForLanding: 100, 1, 1
@@ -2547,6 +2548,7 @@
   hit: 
   callSubroutine: s32'cmnRendaChain_Clear_Idling'
   sprite: s32'bdn202_06', 2
+  headInvul: 0
   recoveryState: 
   sprite: s32'bdn202_07', 2
   sprite: s32'bdn202_08', 2
@@ -2857,9 +2859,9 @@
   sprite: s32'bdn205_09', 2
   sprite: s32'bdn205_10', 2
   HairReset: 1
-  sprite: s32'bdn205_11', 3
-  sprite: s32'bdn205_12', 3
-  sprite: s32'bdn205_13', 3
+  sprite: s32'bdn205_11', 2
+  sprite: s32'bdn205_12', 2
+  sprite: s32'bdn205_13', 2
   endState: 
 
 beginState: s32'NmlAtkAir5A'
@@ -3129,7 +3131,6 @@
   Hair: 5
   HairReset: 1
   sprite: s32'bdn400_01', 1
-  sprite: s32'bdn400_02', 1
   sprite: s32'bdn400_03', 1
   sprite: s32'bdn400_04', 1
   physicsImpulseX: 10000
@@ -3442,7 +3443,6 @@
     setPositionY: 40000
     endIf: 
   sprite: s32'bdn400_20', 1
-  sprite: s32'bdn400_02ex', 1
   sprite: s32'bdn400_03ex', 1
   sprite: s32'bdn400_04ex', 1
   physicsImpulseX: 10000
@@ -3546,9 +3546,9 @@
   HairReset: 4
   haltMomentum: 1
   sprite: s32'bdn400_20', 2
-  sprite: s32'bdn400_02ex', 3
+  sprite: s32'bdn400_02ex', 2
   RandomVoice: s16'vxxx209', 100, s16'vxxx210', 100, s16'', 0, s16'', 0
-  sprite: s32'bdn400_03ex', 3
+  sprite: s32'bdn400_03ex', 1
   sprite: s32'bdn402_00', 1
   createObject: s32'bdn400_circle_wind_airB', (CENTER)
   physicsImpulseX: 6000
@@ -4594,6 +4594,7 @@
   cameraShake: s32'ShakeTate', 500, 0, 30, 5
   createObject: s32'bdn_403_smoke_tame', (PLAYER)
   damage1: 1500
+  blockstunAmount: 22
   sprite: s32'bdn403_33', 3
   sprite: s32'bdn403_31', 3
   sprite: s32'bdn403_32', 3
