--- 1.30/BBS_GKNEF.bbs	2022-06-29 16:26:17.000000000 +0200
+++ 1.31/BBS_GKNEF.bbs	2022-06-29 17:54:18.000000000 +0200
@@ -169,8 +169,11 @@
       ifNot: (VARIABLE), (Result)
         groundHitEffect: (AIR_FACE_UP)
         airHitEffect: (AIR_FACE_UP)
-        hitAirPushbackX: 20000
+        hitAirPushbackX: 30000
         hitAirPushbackY: 20000
+        wallBounceCount: 1
+        wallBounceXVelocityPercent: 100
+        untechAmount: 24
         endIf: 
       endIf: 
     endUpon: 
