--- 1.30/BBS_VTB.bbs	2022-06-29 16:26:28.000000000 +0200
+++ 1.31/BBS_VTB.bbs	2022-06-29 17:54:22.000000000 +0200
@@ -2617,12 +2617,12 @@
   recoveryState: 
   sprite: s32'vtb203_07', 2
   EffectForLanding: 100, 1, 0
-  sprite: s32'vtb203_08', 3
+  sprite: s32'vtb203_08', 6
   sprite: s32'vtb203_09', 4
   HairReset: 1
   haltMomentum: 1
-  sprite: s32'vtb203_10', 3
-  sprite: s32'vtb203_11', 2
+  sprite: s32'vtb203_10', 4
+  sprite: s32'vtb203_11', 3
   endState: 
 
 beginState: s32'NmlAtk2B'
@@ -2895,10 +2895,10 @@
         endIf: 
       endUpon: 
     endUpon: 
-  sprite: s32'vtb205_00', 2
+  sprite: s32'vtb205_00', 1
   HairReset: 1
   sprite: s32'vtb205_01', 2
-  sprite: s32'vtb205_02', 3
+  sprite: s32'vtb205_02', 1
   characterSfx: s32'ARC_BTL_VTB_Atk5D_Bisi'
   randomVoiceLine2: s16'vxxx106', 100, s16'vxxx107', 100, s16'', 0, s16'', 0
   sprite: s32'vtb205_03', 1
@@ -2985,11 +2985,11 @@
         endIf: 
       endUpon: 
     endUpon: 
-  sprite: s32'vtb233_00', 2
+  sprite: s32'vtb233_00', 1
   HairReset: 1
   sprite: s32'vtb233_01', 2
   characterSfx: s32'ARC_BTL_VTB_Sword_Basa'
-  sprite: s32'vtb233_02', 2
+  sprite: s32'vtb233_02', 1
   ForceRandomVoice: s16'vxxx211', 100, s16'vxxx105', 100, s16'vxxx210', 100, s16'', 0
   sprite: s32'vtb233_03', 3
   createObject: s32'vtb_233_tame01', 0
@@ -3080,13 +3080,13 @@
   sprite: s32'vtb205_00', 2
   HairReset: 1
   sprite: s32'vtb205_01', 2
-  sprite: s32'vtb205_02', 3
+  sprite: s32'vtb205_02', 2
   characterSfx: s32'ARC_BTL_VTB_Atk5D_Bisi'
   randomVoiceLine2: s16'vxxx106', 100, s16'vxxx107', 100, s16'', 0, s16'', 0
   sprite: s32'vtb205_03', 1
   sprite: s32'vtb205_04', 1
   sprite: s32'vtb205_05', 1
-  sprite: s32'vtb205_06', 3
+  sprite: s32'vtb205_06', 2
   sprite: s32'vtb205_04', 2
   sprite: s32'vtb205_07', 3
   HairReset: 4
@@ -3474,6 +3474,7 @@
   AtkFrontDirection: 0
   AtkShakeCamera: s32'ShakeYoko', 200, 0, 5, 0
   AtkDamageEffect: 4, s32'cmn_hit_sp3'
+  AtkHoseiGuardBalance: 1000
   upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE)
     clearRegisteredUponCode: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE)
     storeValue: (VARIABLE), 45, (STATIC), 1
