--- 1.30/BBS_CLF.bbs	2022-06-29 16:26:11.000000000 +0200
+++ 1.31/BBS_CLF.bbs	2022-06-29 17:54:16.000000000 +0200
@@ -2334,6 +2334,8 @@
   sprite: s32'clf200_03', 2
   hit: 
   callSubroutine: s32'cmnRendaChain_Clear_Idling'
+  physicsImpulseX: 100000
+  horizontalMomentum: 65
   sprite: s32'clf200_04', 1
   sprite: s32'clf200_04', 1
   recoveryState: 
@@ -2709,7 +2711,7 @@
     endUpon: 
   sprite: s32'clf203_00', 2
   Hair: 5
-  sprite: s32'clf203_01', 2
+  sprite: s32'clf203_01', 1
   sprite: s32'clf203_01', 2
   AttackVoice: 1, -1
   sprite: s32'clf203_02', 2
@@ -2863,7 +2865,7 @@
   sprite: s32'clf204_05', 2
   hit: 
   addPositionX: 50000
-  physicsImpulseX: 100000
+  physicsImpulseX: 140000
   horizontalMomentum: 95
   EffectForDashStart: 104, 1, 0
   createObject: s32'cmn_Windpressure_Yoko', (CENTER)
@@ -2874,12 +2876,12 @@
   sprite: s32'clf204_06', 2
   sprite: s32'clf204_07', 2
   beginLabel: s32'end'
-  sprite: s32'keep', 3
+  sprite: s32'keep', 1
   clearRegisteredUponCode: (ON_ATTACK_HIT_PLAYER_GUARD)
   physicsImpulseX: 50000
   horizontalMomentum: 80
   recoveryState: 
-  sprite: s32'clf204_28', 2
+  sprite: s32'clf204_28', 4
   sprite: s32'clf204_29', 3
   dashStopEffect: 128, 1, 0
   sprite: s32'clf204_30', 3
@@ -3300,8 +3302,8 @@
     whiffCancelOptionBufferTime: s32'NmlAtk3C', 8
     addGatlingOption: s32'NmlAtk3C'
     endUpon: 
-  sprite: s32'clf205_00', 2
-  sprite: s32'clf205_01', 2
+  sprite: s32'clf205_00', 1
+  sprite: s32'clf205_01', 1
   sprite: s32'clf205_02', 2
   randomVoiceLine2: s16'vxxx106', 100, s16'vxxx107', 100, s16'', 0, s16'', 0
   sprite: s32'clf205_03', 2
@@ -5492,6 +5494,9 @@
   triggerUponInObject: s32'Deathflasher_Core', (ON_HIT_OR_BLOCK)
   StopBankSE: 16, 20
   StopBankSE: 17, 20
+  enableWhiffCancel: 1
+  addWhiffCancelOption: s32'CmnActHomingDash'
+  whiffCancelOptionBufferTime: s32'CmnActHomingDash', 10
   sprite: s32'clf401_08', 3
   sprite: s32'clf401_09', 3
   triggerUponInObject: s32'Deathflasher_shadow', (ON_HIT_OR_BLOCK)
@@ -5594,6 +5599,9 @@
   triggerUponInObject: s32'Deathflasher_Core', (ON_HIT_OR_BLOCK)
   StopBankSE: 16, 20
   StopBankSE: 17, 20
+  enableWhiffCancel: 1
+  addWhiffCancelOption: s32'CmnActHomingDash'
+  whiffCancelOptionBufferTime: s32'CmnActHomingDash', 10
   sprite: s32'clf402_08', 3
   sprite: s32'clf402_09', 3
   triggerUponInObject: s32'Deathflasher_shadow', (ON_HIT_OR_BLOCK)
