--- 1.30/BBS_VGB.bbs	2022-06-29 16:26:27.000000000 +0200
+++ 1.31/BBS_VGB.bbs	2022-06-29 17:54:22.000000000 +0200
@@ -2547,7 +2547,7 @@
   sprite: s32'vgb203_13', 2
   sprite: s32'vgb203_14', 3
   sprite: s32'vgb203_15', 3
-  sprite: s32'vgb203_16', 3
+  sprite: s32'vgb203_16', 2
   endState: 
 
 beginState: s32'NmlAtk2B'
@@ -2581,7 +2581,7 @@
   sprite: s32'vgb231_10', 4
   sprite: s32'vgb231_11', 4
   sprite: s32'vgb231_12', 3
-  sprite: s32'vgb231_13', 3
+  sprite: s32'vgb231_13', 2
   endState: 
 
 beginState: s32'NmlAtk6B'
@@ -3876,7 +3876,7 @@
     hitAirPushbackY: 12000
     hitstunAmount: 23
     untechAmount: 26
-    hitStop: 10
+    hitStop: 3
     enemyHitstopAddition: 6, 6, 6
     AtkFrontDirection: 0
     AtkShakeCamera: s32'ShakeYoko', 800, 1, 15, 0
@@ -5140,7 +5140,7 @@
   HairReset: 4
   haltMomentum: 1
   AtkZReversal: 1
-  sprite: s32'vgb403_00', 4
+  sprite: s32'vgb403_00', 2
   physicsImpulseX: 5000
   autoAirTurn: 1
   createObject: s32'vgb_ThirtyPoints_Aura1', (PLAYER)
@@ -5155,6 +5155,10 @@
   sprite: s32'vgb403_01', 3
   physicsImpulseX: 15000
   physicsImpulseY: 25000
+  ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (EnemyDiffY), (STATIC), 200000
+    physicsImpulseY: 5000
+    storeValue: (VARIABLE), 46, (STATIC), 1
+    endIf: 
   resetGravity: 
   sprite: s32'vgb403_02', 3
   physicsImpulseX: 30000
