--- 1.30/BBS_VGSEF.bbs	2022-06-29 16:26:28.000000000 +0200
+++ 1.31/BBS_VGSEF.bbs	2022-06-29 17:54:22.000000000 +0200
@@ -268,7 +268,7 @@
     endUpon: 
   sprite: s32'null', 3
   setDepth: 40000
-  sprite: s32'null', 3
+  sprite: s32'null', 2
   setDepth: -60000
   modifyPositionX: -10000
   modifyPositionY: -10000
@@ -276,7 +276,7 @@
   setDepth: -120000
   modifyPositionX: -60000
   modifyPositionY: 30000
-  sprite: s32'null', 2
+  sprite: s32'null', 1
   setDepth: -100000
   modifyPositionX: 50000
   modifyPositionY: -10000
@@ -666,12 +666,12 @@
     endUpon: 
   sprite: s32'null', 2
   setDepth: -80000
-  sprite: s32'null', 2
+  sprite: s32'null', 1
   setDepth: -110000
   modifyPositionX: -50000
   modifyPositionY: 50000
   sprite: s32'null', 2
-  sprite: s32'null', 2
+  sprite: s32'null', 1
   setDepth: -110000
   modifyPositionX: 10000
   modifyPositionY: 50000
