--- 1.30/BBS_TRSEF.bbs	2022-06-29 16:26:27.000000000 +0200
+++ 1.31/BBS_TRSEF.bbs	2022-06-29 17:54:22.000000000 +0200
@@ -532,14 +532,16 @@
     AtkShakeCamera: s32'ShakeTate', 600, 1, 20, 5
     checkCurrentStateName: 3, s32'CycloneBuster'
     if: (VARIABLE), (Result)
-      hitAirPushbackX: 23000
+      hitAirPushbackX: 13000
       hitAirPushbackY: 30000
-      untechAmount: 26
+      groundHitEffect: 12
+      airHitEffect: 12
+      untechAmount: 36
       enemyHitstopAddition: 0, 3, 3
       IfBeginCoType: 3, 2, 45
         callSubroutine: s32'cmnSpecialAttackEx_ShotInit'
         damage1: 1000
-        untechAmount: 32
+        untechAmount: 36
         enemyHitstopAddition: 0, 5, 5
         endIf: 
       endIf: 
