--- 1.30/BBS_GKN.bbs	2022-06-29 16:26:17.000000000 +0200
+++ 1.31/BBS_GKN.bbs	2022-06-29 17:54:18.000000000 +0200
@@ -3534,13 +3534,13 @@
   horizontalMomentum: 80
   SenkoInputSpecial: 10
   sprite: s32'gkn235_08', 5
-  sprite: s32'gkn235_09', 8
-  sprite: s32'gkn235_10', 5
+  sprite: s32'gkn235_09', 6
+  sprite: s32'gkn235_10', 4
   FaceReset: 20
   HairReset: 1
   haltMomentum: 1
-  sprite: s32'gkn235_11', 5
-  sprite: s32'gkn235_12', 4
+  sprite: s32'gkn235_11', 4
+  sprite: s32'gkn235_12', 3
   sprite: s32'gkn235_13', 3
   endState: 
 
@@ -4962,6 +4962,10 @@
   deactivateObjectByName: s32'GKN_groundSmoke'
   deactivateObjectByName: s32'GKN_kamehameha_shadow'
   triggerUponInObject: s32'GKN_Kamehameha_Core', (ON_HIT_OR_BLOCK)
+  removeWhiffCancelOption: s32'Kamehameha_GKN_Naname'
+  enableWhiffCancel: 1
+  addWhiffCancelOption: s32'CmnActHomingDash'
+  whiffCancelOptionBufferTime: s32'CmnActHomingDash', 15
   sprite: s32'gkn402_14', 3
   sprite: s32'gkn402_16', 3
   sprite: s32'gkn402_17', 3
@@ -5157,6 +5161,10 @@
   clearRegisteredUponCode: (BEAM_CLASH)
   deactivateObjectByName: s32'GKN_kamehameha_shadow'
   triggerUponInObject: s32'GKN_Kamehameha_Core', (ON_HIT_OR_BLOCK)
+  removeWhiffCancelOption: s32'AirKamehameha_GKN_Naname'
+  enableWhiffCancel: 1
+  addWhiffCancelOption: s32'CmnActHomingDash'
+  whiffCancelOptionBufferTime: s32'CmnActHomingDash', 15
   sprite: s32'gkn404_14', 3
   sprite: s32'gkn404_16', 3
   HairReset: 1
@@ -5247,6 +5255,9 @@
   clearRegisteredUponCode: (BEAM_CLASH)
   deactivateObjectByName: s32'GKN_VSLand_kamehameha_shadow'
   triggerUponInObject: s32'GKN_AirKamehameha_Naname_Core', (ON_HIT_OR_BLOCK)
+  enableWhiffCancel: 1
+  addWhiffCancelOption: s32'CmnActHomingDash'
+  whiffCancelOptionBufferTime: s32'CmnActHomingDash', 15
   sprite: s32'gkn405_14', 3
   sprite: s32'gkn405_16', 3
   resetGravity: 
