--- 1.30/BBS_MTN.bbs	2022-06-29 16:26:23.000000000 +0200
+++ 1.31/BBS_MTN.bbs	2022-06-29 17:54:20.000000000 +0200
@@ -3555,7 +3555,7 @@
 beginState: s32'MTN_AirHoming'
   upon: (IMMEDIATE)
     callSubroutine: s32'AirHoming_Init'
-    enableSpecialCancel: 0
+    enableSpecialCancel: 1
     enableSparkCancel: 1
     enableAssistAttack: 1
     enableMemberChangeUltimate: 1
@@ -5661,8 +5661,8 @@
   sprite: s32'mtn204_00', 2
   sprite: s32'mtn204_01', 2
   sprite: s32'mtn204_02', 2
-  sprite: s32'mtn204_03', 2
-  sprite: s32'mtn204_04', 2
+  sprite: s32'mtn204_03', 1
+  sprite: s32'mtn204_04', 1
   AttackVoice: 2, -1
   sprite: s32'mtn204_05', 2
   FaceReset: 503
@@ -6531,6 +6531,7 @@
     AtkFrontDirection: 0
     guardType: (ANY)
     AtkFuttobiFinish: 1
+    untechAmount: 30
     addGatlingOption: s32'NmlAtkAir2D'
     if: (VARIABLE), (IsSparking)
       enableJumpCancel: 1
@@ -7046,13 +7047,13 @@
   dashStopEffect: 128, 1, 1
   sprite: s32'mtn401_55', 3
   HairReset: 2
-  sprite: s32'mtn401_56', 4
+  sprite: s32'mtn401_56', 3
   Hair: 0
   HairReset: 0
-  sprite: s32'mtn401_57', 4
-  sprite: s32'mtn401_58', 4
-  sprite: s32'mtn401_59', 4
-  sprite: s32'mtn401_60', 3
+  sprite: s32'mtn401_57', 3
+  sprite: s32'mtn401_58', 2
+  sprite: s32'mtn401_59', 2
+  sprite: s32'mtn401_60', 2
   endLabel: 
   exitState: 
   beginLabel: s32'guard'
@@ -7116,12 +7117,12 @@
   dashStopEffect: 128, 1, 1
   sprite: s32'mtn401_55', 3
   HairReset: 2
-  sprite: s32'mtn401_56', 4
+  sprite: s32'mtn401_56', 3
   Hair: 0
-  sprite: s32'mtn401_57', 4
-  sprite: s32'mtn401_58', 4
-  sprite: s32'mtn401_59', 4
-  sprite: s32'mtn401_60', 3
+  sprite: s32'mtn401_57', 3
+  sprite: s32'mtn401_58', 2
+  sprite: s32'mtn401_59', 2
+  sprite: s32'mtn401_60', 2
   endLabel: 
   exitState: 
   beginLabel: s32'guard'
@@ -7187,12 +7188,12 @@
   dashStopEffect: 128, 1, 1
   sprite: s32'mtn401_55', 3
   HairReset: 2
-  sprite: s32'mtn401_56', 4
+  sprite: s32'mtn401_56', 3
   Hair: 0
-  sprite: s32'mtn401_57', 4
-  sprite: s32'mtn401_58', 4
-  sprite: s32'mtn401_59', 4
-  sprite: s32'mtn401_60', 3
+  sprite: s32'mtn401_57', 3
+  sprite: s32'mtn401_58', 2
+  sprite: s32'mtn401_59', 2
+  sprite: s32'mtn401_60', 2
   endLabel: 
   exitState: 
   beginLabel: s32'guard'
@@ -7252,14 +7253,14 @@
   horizontalMomentum: 80
   sprite: s32'mtn401_68', 3
   HairReset: 2
-  sprite: s32'mtn401_69', 4
+  sprite: s32'mtn401_69', 3
   Hair: 0
   horizontalMomentum: 100
   resetGravity: 
-  sprite: s32'mtn401_70', 4
-  sprite: s32'mtn401_71', 4
-  sprite: s32'mtn401_72', 4
-  sprite: s32'mtn401_73', 3
+  sprite: s32'mtn401_70', 3
+  sprite: s32'mtn401_71', 2
+  sprite: s32'mtn401_72', 2
+  sprite: s32'mtn401_73', 2
   endLabel: 
   exitState: 
   beginLabel: s32'guard'
@@ -7313,14 +7314,14 @@
   sprite: s32'mtn401_67', 3
   horizontalMomentum: 80
   sprite: s32'mtn401_68', 3
-  sprite: s32'mtn401_69', 4
+  sprite: s32'mtn401_69', 3
   Hair: 0
   horizontalMomentum: 100
   resetGravity: 
-  sprite: s32'mtn401_70', 4
-  sprite: s32'mtn401_71', 4
-  sprite: s32'mtn401_72', 4
-  sprite: s32'mtn401_73', 3
+  sprite: s32'mtn401_70', 3
+  sprite: s32'mtn401_71', 2
+  sprite: s32'mtn401_72', 2
+  sprite: s32'mtn401_73', 2
   endLabel: 
   exitState: 
   beginLabel: s32'guard'
@@ -7378,14 +7379,14 @@
   horizontalMomentum: 80
   sprite: s32'mtn401_68', 3
   HairReset: 2
-  sprite: s32'mtn401_69', 4
+  sprite: s32'mtn401_69', 3
   Hair: 0
   horizontalMomentum: 100
   resetGravity: 
-  sprite: s32'mtn401_70', 4
-  sprite: s32'mtn401_71', 4
-  sprite: s32'mtn401_72', 4
-  sprite: s32'mtn401_73', 3
+  sprite: s32'mtn401_70', 3
+  sprite: s32'mtn401_71', 2
+  sprite: s32'mtn401_72', 2
+  sprite: s32'mtn401_73', 2
   endLabel: 
   exitState: 
   beginLabel: s32'guard'
@@ -8390,6 +8391,8 @@
   hitAirPushbackY: 20000
   enemyHitstopAddition: 0, 3, 3
   AtkRange: 50000000, 0, 0, 0
+  isAirUnblockable: 1
+  damage2: 4
   ifOperation: (IS_EQUAL), (VARIABLE), 45, (STATIC), 10
     damage1: 1000
     hitAirPushbackX: 20000
@@ -8484,6 +8487,7 @@
   superArmorOverdriveGuard: 0
   superArmorBurstGuard: 0
   Unknown1869: 0
+  headInvul: 1
   sprite: s32'mtn235_02', 2
   CommonBankSE: 0, s32'ARC_BTL_CMN_GK_Jump', 100
   AttackVoice: 2, 0
@@ -8507,6 +8511,7 @@
   HairReset: 2
   recoveryState: 
   PushCollisionHeight: -1
+  headInvul: 0
   sprite: s32'mtn235_06', 2
   HairReset: 3
   sprite: s32'mtn235_07ex00', 30
@@ -8516,6 +8521,7 @@
   Hair: 0
   SetCrouch: 1
   enableSuperArmor: 0
+  headInvul: 0
   sprite: s32'mtn235_08', 2
   sprite: s32'mtn235_09', 3
   sprite: s32'mtn235_10', 1
@@ -8538,6 +8544,7 @@
   superArmorOverdriveGuard: 0
   superArmorBurstGuard: 0
   Unknown1869: 0
+  headInvul: 1
   sprite: s32'mtn235_02', 2
   CommonBankSE: 0, s32'ARC_BTL_CMN_GK_Jump', 100
   AttackVoice: 2, 0
@@ -8561,6 +8568,7 @@
   HairReset: 2
   recoveryState: 
   PushCollisionHeight: -1
+  headInvul: 0
   sprite: s32'mtn235_06', 2
   HairReset: 3
   sprite: s32'mtn235_07ex00', 30
@@ -8570,6 +8578,7 @@
   Hair: 0
   SetCrouch: 1
   enableSuperArmor: 0
+  headInvul: 0
   sprite: s32'mtn235_09', 1
   sprite: s32'mtn235_10', 1
   sprite: s32'mtn235_11', 1
@@ -8590,6 +8599,7 @@
   superArmorOverdriveGuard: 0
   superArmorBurstGuard: 0
   Unknown1869: 0
+  headInvul: 1
   sprite: s32'mtn235_01', 1
   sprite: s32'mtn235_01', 2
   sprite: s32'mtn235_02', 2
@@ -8612,6 +8622,7 @@
   HairReset: 2
   recoveryState: 
   PushCollisionHeight: -1
+  headInvul: 0
   sprite: s32'mtn235_06', 2
   HairReset: 3
   sprite: s32'mtn235_07ex00', 30
@@ -8621,6 +8632,7 @@
   Hair: 0
   SetCrouch: 1
   enableSuperArmor: 0
+  headInvul: 0
   sprite: s32'mtn235_09', 1
   sprite: s32'mtn235_10', 1
   sprite: s32'mtn235_11', 1
@@ -8679,6 +8691,17 @@
   clearRegisteredUponCode: (BEAM_CLASH)
   triggerUponInObject: s32'MTN_Kamehameha_Core', (ON_HIT_OR_BLOCK)
   deactivateObjectByName: s32'cmn_kamehameha_shadow'
+  enableWhiffCancel: 1
+  addWhiffCancelOption: s32'Homing'
+  addWhiffCancelOption: s32'Homing1'
+  addWhiffCancelOption: s32'Homing2'
+  addWhiffCancelOption: s32'Homing3'
+  addWhiffCancelOption: s32'Homing4'
+  whiffCancelOptionBufferTime: s32'Homing', 15
+  whiffCancelOptionBufferTime: s32'Homing1', 15
+  whiffCancelOptionBufferTime: s32'Homing2', 15
+  whiffCancelOptionBufferTime: s32'Homing3', 15
+  whiffCancelOptionBufferTime: s32'Homing4', 15
   sprite: s32'mtn403_11', 4
   sprite: s32'mtn403_12', 4
   addPositionX: -40000
