--- 1.30/BBS_SGNEF.bbs	2022-06-29 16:26:26.000000000 +0200
+++ 1.31/BBS_SGNEF.bbs	2022-06-29 17:54:21.000000000 +0200
@@ -1304,7 +1304,7 @@
       hitAirPushbackY: 36000
       pushbackX: 100
       AtkSuikomiYActPosBase: '0x000000006700000000000000000000000000000000000000'
-      enemyHitstopAddition: 0, 5, 5
+      enemyHitstopAddition: 5, 5, 5
       AtkHoseiTadan: 1
       AtkHoseiGuardBalance: 1000
       checkCurrentStateName: 3, s32'AssistAttack'
@@ -2980,6 +2980,7 @@
       airHitEffect: 27
       hitAirPushbackX: 2000
       hitAirPushbackY: 55000
+      Unknown778: 1600
       AtkAdditionalHitCount: 0
       damage1: 513
       minimalDamagePercent: 31
@@ -3025,7 +3026,7 @@
           storeValue: (VARIABLE), 52, (STATIC), 1
           characterSfx: s32'ARC_BTL_SGN_SUGenki_Crush'
           endIf: 
-        verticalMomentum: 95
+        verticalMomentum: 90
         ScreenCollision: 0
         setWallCollision: 0
         deactivateObjectByName: s32'SU_Genki_Smoke00_Eff'
@@ -3063,8 +3064,8 @@
       endUpon: 
     endUpon: 
   sprite: s32'SU_genkidama', 2
-  physicsImpulseX: 14000
-  physicsImpulseY: -20000
+  physicsImpulseX: 19000
+  physicsImpulseY: -40000
   hit: 
   beginLabel: s32'loop'
   sprite: s32'SU_genkidama', 2
