--- 1.30/BBS_VDN.bbs	2022-06-29 16:26:27.000000000 +0200
+++ 1.31/BBS_VDN.bbs	2022-06-29 17:54:22.000000000 +0200
@@ -3033,7 +3033,7 @@
   recoveryState: 
   StopBankSE: 0, 20
   StopBankSE: 1, 30
-  sprite: s32'vdn204_10', 8
+  sprite: s32'vdn204_10', 5
   sprite: s32'vdn204_11', 3
   sprite: s32'vdn204_12', 3
   sprite: s32'vdn204_13', 3
@@ -3198,11 +3198,11 @@
   PushCollisionHeight: -1
   superArmorKiBlastInvul: 0
   sprite: s32'vdn206_30', 2
-  sprite: s32'vdn206_31', 6
+  sprite: s32'vdn206_31', 3
   sprite: s32'vdn206_32', 3
   sprite: s32'vdn206_33', 3
   sprite: s32'vdn206_34', 3
-  sprite: s32'vdn206_35', 3
+  sprite: s32'vdn206_35', 2
   endState: 
 
 beginSubroutine: s32'D_Call_Init'
@@ -3440,11 +3440,11 @@
     endUpon: 
   sprite: s32'vdn263_00', 2
   Hair: 6
-  sprite: s32'vdn263_01', 2
-  sprite: s32'vdn263_02', 2
+  sprite: s32'vdn263_01', 1
+  sprite: s32'vdn263_02', 1
   AttackVoice: 2, -1
   HairReset: 4
-  sprite: s32'vdn263_03', 3
+  sprite: s32'vdn263_03', 2
   sprite: s32'vdn263_04', 2
   cmnSfx: s32'ARC_BTL_CMN_Furi_Hiji'
   sprite: s32'vdn263_05', 1
@@ -5532,7 +5532,7 @@
           endIf: 
         checkCurrentStateName: 23, s32'MoonsaultC'
         if: (VARIABLE), (Result)
-          physicsImpulseX: 30000
+          physicsImpulseX: 45000
           endIf: 
         checkCurrentStateName: 23, s32'AirMoonsaultA'
         if: (VARIABLE), (Result)
@@ -5544,7 +5544,7 @@
           endIf: 
         checkCurrentStateName: 23, s32'AirMoonsaultC'
         if: (VARIABLE), (Result)
-          physicsImpulseX: 35000
+          physicsImpulseX: 45000
           endIf: 
         endElse: 
       endIf: 
@@ -5652,6 +5652,7 @@
   clearRegisteredUponCode: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE)
   upon: (HIT_OR_GUARD_ANY)
     clearRegisteredUponCode: (HIT_OR_GUARD_ANY)
+    enableSpecialCancel: 1
     enableSparkCancel: 1
     enableMemberChangeUltimate: 1
     enableVanish: 1
@@ -6413,6 +6414,7 @@
     tensionUsage: 10000
     FinishCameraRequest: s32'FinishCamera_Rstorm'
     SetRandom: 2, 48, 0, 0, 0, 1
+    headInvul: 1
     endUpon: 
   gotoIfTrue: s32'land', 2, 31
   sprite: s32'vdn022_02', 3
@@ -6517,7 +6519,7 @@
   damage1: 1850
   groundHitEffect: (AIR_LAUNCH)
   airHitEffect: (AIR_LAUNCH)
-  untechAmount: 20
+  untechAmount: 40
   hitStop: 18
   hitAirPushbackX: 11000
   hitAirPushbackY: 75000
@@ -6579,6 +6581,7 @@
   createParticle: s32'cmn_burst_wind', (CENTER)
   createParticle: s32'vdn430_JumpLine', (PLAYER)
   enableSuperArmor: 0
+  headInvul: 0
   if: (VARIABLE), 46
     enableSpecialCancel: 1
     enableJumpCancel: 1
