--- 1.30/BBS_VGBEF.bbs	2022-06-29 16:26:27.000000000 +0200
+++ 1.31/BBS_VGBEF.bbs	2022-06-29 17:54:22.000000000 +0200
@@ -904,7 +904,7 @@
     specialAttack: 
     callSubroutine: s32'cmnAtkLevel_3_AtkInit'
     callSubroutine: s32'cmnSpecialShot_AtkInit'
-    damage1: 1400
+    damage1: 1800
     hitStop: 16
     enemyHitstopAddition: 9, 0, 0
     blockstunAmount: 20
@@ -953,7 +953,7 @@
       endUpon: 
     upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE)
       clearRegisteredUponCode: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE)
-      damage1: 1400
+      damage1: 1800
       endUpon: 
     upon: (ON_ATTACK_COLLISION_SOUSAI)
       ifNot: (VARIABLE), 45
@@ -1094,7 +1094,7 @@
     callSubroutine: s32'cmnAtkLevel_3_AtkInit'
     callSubroutine: s32'cmnSpecialShot_AtkInit'
     callSubroutine: s32'cmn_AssistShotHosei'
-    damage1: 1400
+    damage1: 1800
     hitStop: 16
     enemyHitstopAddition: 9, 0, 0
     blockstunAmount: 20
@@ -1114,7 +1114,7 @@
     SetActionTimeMax: 120
     checkCurrentStateName: 3, s32'AssistAttack'
     if: (VARIABLE), (Result)
-      damage1: 1100
+      damage1: 1600
       untechAmount: 49
       AtkHoseiTadan: 1
       callSubroutine: s32'cmn_AssistHosei'
@@ -1153,10 +1153,10 @@
       endUpon: 
     upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE)
       clearRegisteredUponCode: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE)
-      damage1: 1400
+      damage1: 1800
       checkCurrentStateName: 3, s32'AssistAttack'
       if: (VARIABLE), (Result)
-        damage1: 1100
+        damage1: 1600
         endIf: 
       endUpon: 
     upon: (ON_ATTACK_COLLISION_SOUSAI)
