--- 1.30/BBS_KRN.bbs	2022-06-29 16:26:21.000000000 +0200
+++ 1.31/BBS_KRN.bbs	2022-06-29 17:54:19.000000000 +0200
@@ -4117,7 +4117,6 @@
   dashStopEffect: 100, 1, 1
   StopBankSE: 0, 20
   sprite: s32'krn036_03', 3
-  callSubroutine: s32'cmn_NandemoEnableStand'
   sprite: s32'krn036_02', 3
   sprite: s32'krn036_01', 3
   sprite: s32'krn036_00', 3
@@ -4403,6 +4402,8 @@
   createObject: s32'AirZanzou_C', (PLAYER)
   StackControlObject: 8, 1
   sprite: s32'krn400_30', 3
+  addVelocityY: -12000
+  setGravity: 4000
   sprite: s32'krn400_31', 3
   sprite: s32'krn022_04', 3
   sprite: s32'krn022_05', 2
@@ -4599,11 +4600,11 @@
   createObject: s32'Senzu', 0
   ForceRandomVoice: s16'vxxx204', 100, s16'', 0, s16'', 0, s16'', 0
   sprite: s32'krn401_09', 2
-  sprite: s32'krn401_10', 3
-  sprite: s32'krn401_13', 2
+  sprite: s32'krn401_10', 1
+  sprite: s32'krn401_13', 1
   resetGravity: 
-  sprite: s32'krn401_14', 2
-  sprite: s32'krn401_15', 2
+  sprite: s32'krn401_14', 1
+  sprite: s32'krn401_15', 1
   endState: 
 
 beginState: s32'AirSenzu_A'
@@ -4724,15 +4725,15 @@
   ForceRandomVoice: s16'vxxx204', 100, s16'', 0, s16'', 0, s16'', 0
   jumpToStateInObject: (PARENT), s32'landing'
   sprite: s32'krn401_09', 2
-  sprite: s32'krn401_10', 3
-  sprite: s32'krn401_11', 3
+  sprite: s32'krn401_10', 2
+  sprite: s32'krn401_11', 2
   resetGravity: 
-  sprite: s32'krn401_12', 3
-  sprite: s32'krn401_10', 3
-  sprite: s32'krn401_11', 3
-  sprite: s32'krn401_13', 2
-  sprite: s32'krn401_14', 2
-  sprite: s32'krn401_15', 2
+  sprite: s32'krn401_12', 2
+  sprite: s32'krn401_10', 1
+  sprite: s32'krn401_11', 1
+  sprite: s32'krn401_13', 1
+  sprite: s32'krn401_14', 1
+  sprite: s32'krn401_15', 1
   endLabel: 
   exitState: 
   beginLabel: s32'landing'
@@ -5523,6 +5524,9 @@
   deactivateObjectByName: s32'cmn_GroundSmoke'
   sprite: s32'krn402_13', 3
   sprite: s32'krn402_14', 3
+  enableWhiffCancel: 1
+  addWhiffCancelOption: s32'CmnActHomingDash'
+  whiffCancelOptionBufferTime: s32'CmnActHomingDash', 10
   endLabel: 
   gotoIfFalse: s32'end_land2', 2, 52
   sprite: s32'krn402_12', 2
@@ -5606,6 +5610,9 @@
   haltMomentum: 1
   triggerUponInObject: s32'Kamehameha_Core', (ON_HIT_OR_BLOCK)
   deactivateObjectByName: s32'Kamehameha_Shadow'
+  enableWhiffCancel: 1
+  addWhiffCancelOption: s32'CmnActHomingDash'
+  whiffCancelOptionBufferTime: s32'CmnActHomingDash', 10
   sprite: s32'krn402_31', 3
   sprite: s32'krn402_32', 3
   endLabel: 
