--- 1.30/BBS_BRS.bbs	2022-06-29 16:26:09.000000000 +0200
+++ 1.31/BBS_BRS.bbs	2022-06-29 17:54:16.000000000 +0200
@@ -2562,8 +2562,8 @@
   sprite: s32'brs230_04', 4
   recoveryState: 
   enableWhiffCancel: 1
-  sprite: s32'brs230_05', 4
-  sprite: s32'brs230_06', 4
+  sprite: s32'brs230_05', 3
+  sprite: s32'brs230_06', 3
   enableWhiffCancel: 0
   sprite: s32'brs230_07', 3
   endState: 
@@ -2583,17 +2583,17 @@
     jumpToStateInObject: 9, s32'end'
     callSubroutine: s32'SuperArmor_Normal'
     endUpon: 
-  sprite: s32'brs203_00', 2
+  sprite: s32'brs203_00', 3
   Hair: 5
   HairReset: 4
-  sprite: s32'brs203_01', 2
-  sprite: s32'brs203_02', 2
-  physicsImpulseX: 65000
+  sprite: s32'brs203_01', 4
+  sprite: s32'brs203_02', 1
+  physicsImpulseX: 100000
   cmnSfx: s32'ARC_BTL_CMN_GK_Jump'
-  sprite: s32'brs203_03', 2
-  horizontalMomentum: 90
+  sprite: s32'brs203_03', 1
+  horizontalMomentum: 85
   AttackVoice: 1, -1
-  sprite: s32'brs203_04', 2
+  sprite: s32'brs203_04', 1
   cmnSfx: s32'ARC_BTL_CMN_Furi_SyagamiKeri'
   sprite: s32'brs203_05', 6
   callSubroutine: s32'cmnRendaChain_Clear_Idling'
@@ -2604,6 +2604,7 @@
   sprite: s32'brs203_06', 3
   callSubroutine: s32'SuperArmor_End'
   recoveryState: 
+  horizontalMomentum: 70
   sprite: s32'brs203_07', 4
   sprite: s32'brs203_08', 4
   dashStopEffect: 104, 1, 1
@@ -2658,13 +2659,13 @@
   recoveryState: 
   sprite: s32'brs231_08', 2
   extendPushboxX: 0
-  sprite: s32'brs231_09', 3
+  sprite: s32'brs231_09', 6
   sprite: s32'brs231_10', 3
   sprite: s32'brs231_11', 3
-  sprite: s32'brs231_12', 2
-  sprite: s32'brs231_13', 2
+  sprite: s32'brs231_12', 3
+  sprite: s32'brs231_13', 3
   pushboxHeightCrouch: 220000
-  sprite: s32'brs231_14', 2
+  sprite: s32'brs231_14', 3
   endState: 
 
 beginState: s32'NmlAtk6B'
@@ -2926,10 +2927,10 @@
   sprite: s32'brs205_04', 3
   sprite: s32'brs205_05', 3
   storeValue: (VARIABLE), 47, (STATIC), 0
-  sprite: s32'brs205_06', 3
-  sprite: s32'brs205_07', 3
-  sprite: s32'brs205_09', 3
-  sprite: s32'brs205_10', 3
+  sprite: s32'brs205_06', 5
+  sprite: s32'brs205_07', 5
+  sprite: s32'brs205_09', 4
+  sprite: s32'brs205_10', 4
   endState: 
 
 beginState: s32'NmlAtkAir5A'
@@ -2943,7 +2944,7 @@
     endUpon: 
   sprite: s32'brs260_00', 1
   sprite: s32'brs260_01', 1
-  sprite: s32'brs260_02', 2
+  sprite: s32'brs260_02', 1
   sprite: s32'brs260_03', 1
   AttackVoice: 0, -1
   cmnSfx: s32'ARC_BTL_CMN_Furi_Hook'
@@ -3573,6 +3574,7 @@
       clearRegisteredUponCode: (HIT_OR_GUARD)
       if: (VARIABLE), 56
         damage1: 1400
+        pushbackX: 350
         endIf: 
       endUpon: 
     endIf: 
@@ -3584,6 +3586,7 @@
       clearRegisteredUponCode: (HIT_OR_GUARD)
       if: (VARIABLE), 56
         damage1: 1500
+        pushbackX: 350
         endIf: 
       endUpon: 
     endIf: 
@@ -4070,6 +4073,9 @@
   clearRegisteredUponCode: (FRAME_STEP)
   physicsImpulseX: 40000
   horizontalMomentum: 90
+  if: (VARIABLE), 56
+    physicsImpulseX: 60000
+    endIf: 
   characterSfx: s32'ARC_BTL_BRS_GClaw_Furi_Ctch'
   sprite: s32'brs401_06', 3
   createObject: s32'brs_401_eff01', (PLAYER)
@@ -4121,6 +4127,9 @@
   sprite: s32'brs401_05', 3
   clearRegisteredUponCode: (FRAME_STEP)
   physicsImpulseX: 80000
+  if: (VARIABLE), 56
+    physicsImpulseX: 120000
+    endIf: 
   horizontalMomentum: 90
   characterSfx: s32'ARC_BTL_BRS_GClaw_Furi_Ctch'
   sprite: s32'brs401_06', 4
@@ -4466,6 +4475,9 @@
   physicsImpulseY: 70000
   horizontalMomentum: 90
   verticalMomentum: 90
+  if: (VARIABLE), 56
+    physicsImpulseX: 125000
+    endIf: 
   hit: 
   sprite: s32'brs402_17', 3
   verticalMomentum: 100
@@ -4515,6 +4527,9 @@
   physicsImpulseY: 70000
   horizontalMomentum: 90
   verticalMomentum: 90
+  if: (VARIABLE), 56
+    physicsImpulseX: 180000
+    endIf: 
   hit: 
   upon: (FRAME_STEP)
     ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (EnemyFrontDistance), (STATIC), 500000
