--- 1.30/BBS_GTL.bbs	2022-06-29 16:26:19.000000000 +0200
+++ 1.31/BBS_GTL.bbs	2022-06-29 17:54:19.000000000 +0200
@@ -2307,7 +2307,7 @@
     sprite: s32'gtl201_00', 2
     FaceReset: 22
     HairReset: 4
-    sprite: s32'gtl201_01', 3
+    sprite: s32'gtl201_01', 2
     AttackVoice: 0, -1
     sprite: s32'gtl201_02', 2
     sprite: s32'gtl201_03', 2
@@ -2389,6 +2389,7 @@
     physicsImpulseY: 20000
     resetGravity: 
     recoveryState: 
+    deactivateObjectByName: s32'GTL_5A3rdEff'
     sprite: s32'gtl100_04ex00', 2
     sprite: s32'gtl100_05ex00', 2
     sprite: s32'gtl022_04', 3
@@ -2452,7 +2453,7 @@
     sprite: s32'gtl203_11', 3
     horizontalMomentum: 70
     sprite: s32'gtl203_12', 3
-    sprite: s32'gtl203_13', 3
+    sprite: s32'gtl203_13', 2
     physicsImpulseX: 0
     endState: 
   beginState: s32'NmlAtk2B'
@@ -4254,25 +4255,25 @@
               endElse: 
             endIf: 
           ifOperation: (IS_EQUAL), (VARIABLE), 47, (STATIC), 6
-            ifOperation: (IS_LESSER), (VARIABLE), (SpeedX), (STATIC), 25000
-              addVelocityX: 2500
+            ifOperation: (IS_LESSER), (VARIABLE), (SpeedX), (STATIC), 50000
+              addVelocityX: 7500
               endIf: 
-            ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), (SpeedX), (STATIC), 25000
-              physicsImpulseX: 25000
+            ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), (SpeedX), (STATIC), 50000
+              physicsImpulseX: 50000
               endIf: 
             endIf: 
           ifOperation: (IS_EQUAL), (VARIABLE), 47, (STATIC), 4
-            ifOperation: (IS_GREATER), (VARIABLE), (SpeedX), (STATIC), -25000
-              addVelocityX: -2500
+            ifOperation: (IS_GREATER), (VARIABLE), (SpeedX), (STATIC), -50000
+              addVelocityX: -7500
               endIf: 
-            ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (SpeedX), (STATIC), -25000
-              physicsImpulseX: -25000
+            ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (SpeedX), (STATIC), -50000
+              physicsImpulseX: -50000
               endIf: 
             endIf: 
           checkInput: (INPUT_ANY_UP)
           if: (VARIABLE), (Result)
             ifOperation: (IS_LESSER), (VARIABLE), (SpeedY), (STATIC), 25000
-              addVelocityY: 2500
+              addVelocityY: 5000
               endIf: 
             ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), (SpeedY), (STATIC), 25000
               physicsImpulseY: 25000
@@ -4281,7 +4282,7 @@
           checkInput: (INPUT_ANY_DOWN)
           if: (VARIABLE), (Result)
             ifOperation: (IS_GREATER), (VARIABLE), (SpeedY), (STATIC), -25000
-              addVelocityY: -2500
+              addVelocityY: -5000
               endIf: 
             ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (SpeedY), (STATIC), -25000
               physicsImpulseY: -25000
@@ -4369,25 +4370,25 @@
               endElse: 
             endIf: 
           ifOperation: (IS_EQUAL), (VARIABLE), 47, (STATIC), 6
-            ifOperation: (IS_LESSER), (VARIABLE), (SpeedX), (STATIC), 25000
-              addVelocityX: 2500
+            ifOperation: (IS_LESSER), (VARIABLE), (SpeedX), (STATIC), 50000
+              addVelocityX: 7500
               endIf: 
-            ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), (SpeedX), (STATIC), 25000
-              physicsImpulseX: 25000
+            ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), (SpeedX), (STATIC), 50000
+              physicsImpulseX: 50000
               endIf: 
             endIf: 
           ifOperation: (IS_EQUAL), (VARIABLE), 47, (STATIC), 4
-            ifOperation: (IS_GREATER), (VARIABLE), (SpeedX), (STATIC), -25000
-              addVelocityX: -2500
+            ifOperation: (IS_GREATER), (VARIABLE), (SpeedX), (STATIC), -50000
+              addVelocityX: -7500
               endIf: 
-            ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (SpeedX), (STATIC), -25000
-              physicsImpulseX: -25000
+            ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (SpeedX), (STATIC), -50000
+              physicsImpulseX: -50000
               endIf: 
             endIf: 
           checkInput: (INPUT_ANY_UP)
           if: (VARIABLE), (Result)
             ifOperation: (IS_LESSER), (VARIABLE), (SpeedY), (STATIC), 25000
-              addVelocityY: 2500
+              addVelocityY: 5000
               endIf: 
             ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), (SpeedY), (STATIC), 25000
               physicsImpulseY: 25000
@@ -4396,7 +4397,7 @@
           checkInput: (INPUT_ANY_DOWN)
           if: (VARIABLE), (Result)
             ifOperation: (IS_GREATER), (VARIABLE), (SpeedY), (STATIC), -25000
-              addVelocityY: -2500
+              addVelocityY: -5000
               endIf: 
             ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (SpeedY), (STATIC), -25000
               physicsImpulseY: -25000
@@ -5292,6 +5293,9 @@
     triggerUponInObject: s32'Revengeshout_core', (ON_HIT_OR_BLOCK)
     StopBankSE: 16, 20
     StopBankSE: 17, 20
+    enableWhiffCancel: 1
+    addWhiffCancelOption: s32'CmnActHomingDash'
+    whiffCancelOptionBufferTime: s32'CmnActHomingDash', 10
     sprite: s32'gtl404_12', 3
     sprite: s32'gtl404_13', 3
     endLabel: 
@@ -5612,6 +5616,9 @@
     sprite: s32'gtl430_26', 3
     sprite: s32'gtl430_27', 3
     sprite: s32'gtl430_25', 3
+    sprite: s32'gtl430_26', 3
+    sprite: s32'gtl430_27', 3
+    sprite: s32'gtl430_25', 3
     sprite: s32'gtl430_26', 3
     sprite: s32'gtl430_27', 3
     sprite: s32'gtl430_25', 3
