--- 1.30/BBS_JRN.bbs	2022-06-29 16:26:20.000000000 +0200
+++ 1.31/BBS_JRN.bbs	2022-06-29 17:54:19.000000000 +0200
@@ -2187,8 +2187,6 @@
     callSubroutine: s32'cmnNmlAtk5A_AtkInit'
     callSubroutine: s32'cmnNmlAtk5A_Init'
     addGatlingOption: s32'NmlAtk2D'
-    removeGatlingOption: s32'NmlAtk5A'
-    removeWhiffCancelOption: s32'NmlAtk5A'
     endUpon: 
   sprite: s32'jrn200_00', 1
   Hair: 5
@@ -2563,7 +2561,7 @@
     endUpon: 
   sprite: s32'jrn203_00', 1
   Hair: 5
-  sprite: s32'jrn203_01', 4
+  sprite: s32'jrn203_01', 2
   sprite: s32'jrn203_02', 1
   sprite: s32'jrn203_03', 1
   physicsImpulseX: 110000
@@ -2627,8 +2625,8 @@
     endUpon: 
   sprite: s32'jrn231_00', 2
   Hair: 5
-  sprite: s32'jrn231_01', 2
-  sprite: s32'jrn231_02', 2
+  sprite: s32'jrn231_01', 1
+  sprite: s32'jrn231_02', 1
   AttackVoice: 1, 112
   sprite: s32'jrn231_03', 2
   cmnSfx: s32'ARC_BTL_CMN_Furi_SyagamiKeri'
@@ -3958,10 +3956,10 @@
   upon: (IMMEDIATE)
     callSubroutine: s32'Justice_Init'
     endUpon: 
-  sprite: s32'jrn400_00', 2
+  sprite: s32'jrn400_00', 1
   Hair: 5
-  sprite: s32'jrn400_01', 2
-  sprite: s32'jrn400_02', 2
+  sprite: s32'jrn400_01', 1
+  sprite: s32'jrn400_02', 1
   sprite: s32'jrn400_03', 2
   characterSfx: s32'ARC_BTL_JRN_Justice_Bishi'
   RandomVoice: s16'vxxx200', 100, s16'vxxx201', 100, s16'vxxx202', 100, s16'', 0
@@ -4071,11 +4069,11 @@
   upon: (IMMEDIATE)
     callSubroutine: s32'Justice_Init'
     endUpon: 
-  sprite: s32'jrn401_00', 2
+  sprite: s32'jrn401_00', 1
   haltMomentum: 1
   Hair: 5
-  sprite: s32'jrn401_01', 2
-  sprite: s32'jrn401_02', 2
+  sprite: s32'jrn401_01', 1
+  sprite: s32'jrn401_02', 1
   sprite: s32'jrn401_03', 2
   characterSfx: s32'ARC_BTL_JRN_Justice_Bishi'
   RandomVoice: s16'vxxx200', 100, s16'vxxx201', 100, s16'vxxx202', 100, s16'', 0
@@ -4313,13 +4311,13 @@
   SetCrouch: 1
   Hair: 5
   createObject: s32'jrn402_Smokes_Eff', (PLAYER)
-  sprite: s32'jrn010_01', 1
-  sprite: s32'jrn402_24', 1
   callSubroutine: s32'cmn_MutekiArmor'
   NoGuardEffect: 0
   superArmorLoseMid: 0
   superArmorLoseHigh: 0
   superArmorHitstop: 5, -2
+  sprite: s32'jrn010_01', 1
+  sprite: s32'jrn402_24', 1
   sprite: s32'jrn402_25', 2
   characterSfx: s32'ARC_BTL_JRN_Shock_Bishi'
   RandomVoice: s16'vxxx203', 100, s16'vxxx205', 100, s16'', 0, s16'', 0
@@ -4492,8 +4490,8 @@
   RandomVoice: s16'vxxx213', 100, s16'vxxx214', 100, s16'', 0, s16'', 0
   sprite: s32'jrn404_02', 2
   sprite: s32'jrn404_03', 2
-  sprite: s32'jrn404_04', 3
-  sprite: s32'jrn404_05', 2
+  sprite: s32'jrn404_04', 2
+  sprite: s32'jrn404_05', 1
   sprite: s32'jrn404_06', 1
   physicsImpulseX: 160000
   horizontalMomentum: 80
