--- 1.30/BBS_FRN.bbs	2022-06-29 16:26:13.000000000 +0200
+++ 1.31/BBS_FRN.bbs	2022-06-29 17:54:17.000000000 +0200
@@ -154,7 +154,7 @@
     characterState: (CROUCHING)
     moveInput: (INPUT_PRESS_KI)
     moveInput: (INPUT_1)
-    move_ShareChain: s32'NmlAtk2D'
+    maxChainIntoSelf: 1
     SkillEstAtkBox: -400, -800, 300, 600
     SkillEstTagNoCombo: 
     SkillEstPoint: 30
@@ -178,7 +178,7 @@
     characterState: (CROUCHING)
     moveInput: (INPUT_PRESS_KI)
     moveInput: (INPUT_3)
-    move_ShareChain: s32'NmlAtk2D'
+    maxChainIntoSelf: 1
     SkillEstAtkBox: -800, -400, 200, 200
     SkillEstTagNoCombo: 
     SkillEstPoint: 30
@@ -3169,8 +3169,8 @@
   HairReset: 4
   createObject: s32'frn232_BodyAuraEff', (PLAYER)
   createObject: s32'frn232_HandAuraEff', (PLAYER)
-  sprite: s32'frn232_01', 2
-  sprite: s32'frn232_02', 2
+  sprite: s32'frn232_01', 1
+  sprite: s32'frn232_02', 1
   SetCrouch: 0
   AttackVoice: 2, -1
   sprite: s32'frn232_03', 2
@@ -3245,8 +3245,8 @@
   Hair: 4
   HairReset: 4
   sprite: s32'frn205_01', 1
-  sprite: s32'frn205_02', 2
-  sprite: s32'frn205_03', 2
+  sprite: s32'frn205_02', 1
+  sprite: s32'frn205_03', 1
   SmartVoice: s16'vxxx106'
   sprite: s32'frn205_04', 1
   sprite: s32'frn205_05', 1
@@ -3374,6 +3374,8 @@
     addWhiffCancelOption: s32'NmlAtk5C'
     addWhiffCancelOption: s32'NmlAtk2C'
     addWhiffCancelOption: s32'NmlAtk5D'
+    addWhiffCancelOption: s32'NmlAtk2D'
+    addWhiffCancelOption: s32'NmlAtk3D'
     addWhiffCancelOption: s32'WarpSmash'
     addWhiffCancelOption: s32'DeathWave_A'
     addWhiffCancelOption: s32'DeathWave_B'
@@ -3403,12 +3405,12 @@
   Hair: 5
   HairReset: 4
   sprite: s32'frn233_01', 1
-  sprite: s32'frn233_03', 2
+  sprite: s32'frn233_03', 1
   sprite: s32'frn233_04', 1
-  sprite: s32'frn233_05', 2
+  sprite: s32'frn233_05', 1
   SetCrouch: 0
   physicsImpulseX: -15000
-  sprite: s32'frn233_06', 2
+  sprite: s32'frn233_06', 1
   sprite: s32'frn233_07', 2
   sprite: s32'frn233_08', 2
   SmartVoice: s16'vxxx107'
@@ -3441,6 +3443,8 @@
     addWhiffCancelOption: s32'NmlAtk5C'
     addWhiffCancelOption: s32'NmlAtk2C'
     addWhiffCancelOption: s32'NmlAtk5D'
+    addWhiffCancelOption: s32'NmlAtk1D'
+    addWhiffCancelOption: s32'NmlAtk3D'
     addWhiffCancelOption: s32'WarpSmash'
     addWhiffCancelOption: s32'DeathWave_A'
     addWhiffCancelOption: s32'DeathWave_B'
@@ -3470,12 +3474,12 @@
   Hair: 5
   HairReset: 4
   sprite: s32'frn233_01', 1
-  sprite: s32'frn233_03', 2
+  sprite: s32'frn233_03', 1
   sprite: s32'frn233_04', 1
-  sprite: s32'frn233_05', 2
+  sprite: s32'frn233_05', 1
   SetCrouch: 0
   physicsImpulseX: -15000
-  sprite: s32'frn233_06', 2
+  sprite: s32'frn233_06', 1
   sprite: s32'frn233_07', 2
   sprite: s32'frn233_08', 2
   SmartVoice: s16'vxxx107'
@@ -3508,6 +3512,8 @@
     addWhiffCancelOption: s32'NmlAtk5C'
     addWhiffCancelOption: s32'NmlAtk2C'
     addWhiffCancelOption: s32'NmlAtk5D'
+    addWhiffCancelOption: s32'NmlAtk1D'
+    addWhiffCancelOption: s32'NmlAtk2D'
     addWhiffCancelOption: s32'WarpSmash'
     addWhiffCancelOption: s32'DeathWave_A'
     addWhiffCancelOption: s32'DeathWave_B'
@@ -3537,12 +3543,12 @@
   Hair: 5
   HairReset: 4
   sprite: s32'frn233_01', 1
-  sprite: s32'frn233_03', 2
+  sprite: s32'frn233_03', 1
   sprite: s32'frn233_04', 1
-  sprite: s32'frn233_05', 2
+  sprite: s32'frn233_05', 1
   SetCrouch: 0
   physicsImpulseX: 15000
-  sprite: s32'frn233_06', 2
+  sprite: s32'frn233_06', 1
   sprite: s32'frn233_07', 2
   sprite: s32'frn233_08', 2
   SmartVoice: s16'vxxx107'
@@ -3874,11 +3880,11 @@
   Hair: 3
   HairReset: 4
   haltMomentum: 1
-  sprite: s32'frn265_01', 2
+  sprite: s32'frn265_01', 1
   sprite: s32'frn265_02', 1
-  sprite: s32'frn265_03', 2
+  sprite: s32'frn265_03', 1
   sprite: s32'frn265_04', 1
-  sprite: s32'frn265_05', 2
+  sprite: s32'frn265_05', 1
   SmartVoice: s16'vxxx102'
   sprite: s32'frn265_06', 2
   cmnSfx: s32'ARC_BTL_CMN_Furi_Hook'
@@ -4137,11 +4143,10 @@
   sprite: s32'frn400_10', 4
   sprite: s32'frn400_11', 4
   sprite: s32'frn400_09', 4
-  sprite: s32'frn400_10', 4
   spriteAddTime: 2, 2, 46
-  sprite: s32'frn400_12', 3
-  sprite: s32'frn400_13', 3
-  sprite: s32'frn400_14', 3
+  sprite: s32'frn400_12', 2
+  sprite: s32'frn400_13', 2
+  sprite: s32'frn400_14', 2
   endState: 
 
 beginState: s32'DeathWave_C'
@@ -4248,13 +4253,10 @@
   setGravity: 1000
   sprite: s32'frn402_14', 3
   sprite: s32'frn402_15', 3
-  sprite: s32'frn402_16', 3
+  sprite: s32'frn402_16', 1
   resetGravity: 
-  sprite: s32'frn402_17', 3
-  sprite: s32'frn402_18', 3
-  sprite: s32'frn402_19', 3
-  sprite: s32'frn402_20', 3
-  sprite: s32'frn022_05', 3
+  landingRecovery: 6, 0
+  sprite: s32'frn022_05', 2
   endLabel: 
   exitState: 
   beginLabel: s32'sousai'
@@ -4458,9 +4460,6 @@
   sprite: s32'frn402_15', 3
   sprite: s32'frn402_16', 3
   sprite: s32'frn402_17', 3
-  sprite: s32'frn402_18', 3
-  sprite: s32'frn402_19', 3
-  sprite: s32'frn402_20', 3
   sprite: s32'frn022_05', 3
   endLabel: 
   exitState: 
