--- 1.30/BBS_BSN.bbs	2022-06-29 16:26:09.000000000 +0200
+++ 1.31/BBS_BSN.bbs	2022-06-29 17:54:16.000000000 +0200
@@ -138,7 +138,7 @@
     characterState: (STANDING)
     moveInput: (INPUT_PRESS_KI)
     moveInput: (INPUT_DISALLOW_UP)
-    maxChainIntoSelf: 1
+    maxChainIntoSelf: 2
     SkillEstAtkBox: -1500, -250, 1500, 250
     SkillEstLevelLimit: 30
     SkillEstTagGuard: 
@@ -2965,7 +2965,7 @@
   CommonBankSE: 0, s32'ARC_BTL_CMN_JetDashStart', 100
   sprite: s32'bsn231_04', 2
   createParticle: s32'bg_dash_default', (PLAYER)
-  physicsImpulseX: 40000
+  physicsImpulseX: 45000
   horizontalMomentum: 107
   sprite: s32'bsn231_05', 3
   createParticle: s32'bg_dash_default', (PLAYER)
@@ -3207,6 +3207,12 @@
     callSubroutine: s32'cmnNmlAtk2D_Init'
     addWhiffCancelOption: s32'NmlAtk2D_Chain'
     whiffCancelOptionBufferTime: s32'NmlAtk2D_Chain', 8
+    addWhiffCancelOption: s32'NmlAtk5C'
+    whiffCancelOptionBufferTime: s32'NmlAtk5C', 8
+    addWhiffCancelOption: s32'NmlAtk2C'
+    whiffCancelOptionBufferTime: s32'NmlAtk2C', 8
+    addWhiffCancelOption: s32'NmlAtk5D'
+    whiffCancelOptionBufferTime: s32'NmlAtk5D', 8
     endUpon: 
   sprite: s32'bsn233_00', 2
   FaceReset: 501
@@ -4801,8 +4807,6 @@
   sprite: s32'bsn403_04', 3
   HairReset: 4
   sprite: s32'bsn403_05', 3
-  sprite: s32'bsn403_06', 3
-  sprite: s32'bsn403_04', 3
   sprite: s32'bsn403_07', 1
   characterSfx: s32'ARC_BTL_BSN_seisei_BallBorn'
   triggerUponInObject: s32'hakaiShot', 32
