--- 1.30/BBS_AEN.bbs	2022-06-29 16:26:08.000000000 +0200
+++ 1.31/BBS_AEN.bbs	2022-06-29 17:54:15.000000000 +0200
@@ -2558,6 +2558,7 @@
   upon: (IMMEDIATE)
     callSubroutine: s32'cmnNmlAtk5A3rd_AtkInit'
     callSubroutine: s32'cmnNmlAtk5A3rd_Init'
+    pushbackX: 20
     upon: 49
       CheckSignalName: s32'CameraCombo_5A3rd'
       if: (VARIABLE), (Result)
@@ -2651,7 +2652,7 @@
     addGatlingOption: s32'NmlAtk2D'
     pushbackX: 120
     upon: (ON_ATTACK_HIT_PLAYER_GUARD)
-      PerSpeedX: 10
+      PerSpeedX: 80
       endUpon: 
     endUpon: 
   sprite: s32'aen203_00', 1
@@ -2664,7 +2665,7 @@
   AttackVoice: 1, -1
   cmnSfx: s32'ARC_BTL_CMN_Furi_Hiji'
   sprite: s32'aen203_03', 1
-  physicsImpulseX: 30000
+  physicsImpulseX: 50000
   sprite: s32'aen203_06', 2
   PerSpeedX: 70
   sprite: s32'aen203_07', 2
@@ -2673,7 +2674,7 @@
   horizontalMomentum: 95
   sprite: s32'aen203_08', 2
   sprite: s32'aen203_09', 2
-  horizontalMomentum: 60
+  horizontalMomentum: 80
   recoveryState: 
   sprite: s32'aen203_10', 3
   sprite: s32'aen203_11', 4
@@ -3836,6 +3837,12 @@
     if: (VARIABLE), (EnemyInHitstun)
       IfBeginCoType: 32, 2, 30
         CreateArg_HikitsugiVal: 0, 1, 0, 0
+        ifOperation: (IS_LESSER), (VARIABLE), (EnemyFrontDistance), (STATIC), 400000
+          exPointFReset: (ENEMY_LAND), 0, 0
+          endIf: 
+        else: 
+          exPointFReset: (PLAYER_LAND), 400000, 0
+          endElse: 
         endIf: 
       endIf: 
     createObjectWithArg: s32'17_senkyuutai', (EX_POINT_F)
@@ -4051,7 +4058,9 @@
   PushSpeedY: 
   haltMomentum: 1
   sprite: s32'aen403_01', 2
+  spriteAddTime: 1, 2, 214
   sprite: s32'aen403_02', 2
+  spriteAddTime: 1, 2, 214
   voiceLine: s16'vxxx204'
   HairReset: 4
   sprite: s32'aen403_03', 2
@@ -4288,9 +4297,9 @@
   sprite: s32'aen404_03', 1
   sprite: s32'aen404_04', 1
   ForceRandomVoiceWeak: s16'vxxx209', 100, s16'vxxx210', 100, s16'', 0, s16'', 0
-  sprite: s32'aen404_05', 2
+  sprite: s32'aen404_05', 1
   createObject: s32'kienzan_start', 0
-  sprite: s32'aen404_06', 2
+  sprite: s32'aen404_06', 1
   sprite: s32'aen404_07', 2
   sprite: s32'aen404_08', 2
   sprite: s32'aen404_09', 2
