--- 1.30/BBS_ZMB.bbs	2022-06-29 16:26:29.000000000 +0200
+++ 1.31/BBS_ZMB.bbs	2022-06-29 17:54:23.000000000 +0200
@@ -2218,8 +2218,6 @@
   upon: (IMMEDIATE)
     callSubroutine: s32'cmnNmlAtk5A_AtkInit'
     callSubroutine: s32'cmnNmlAtk5A_Init'
-    removeGatlingOption: s32'NmlAtk5A'
-    removeWhiffCancelOption: s32'NmlAtk5A'
     AtkDamageEffect: 4, s32'cmn_slash_s'
     AtkHitCommonSEReset: s32'ARC_BTL_VTB_Atk5D_Hit'
     guardSoundEffect: s32''
@@ -2586,8 +2584,10 @@
   addPositionX: 20000
   sprite: s32'zmb204_04', 3
   addPositionX: 20000
+  physicsImpulseX: 20000
   sprite: s32'zmb204_05', 2
   cmnSfx: s32'ARC_BTL_CMN_Furi_YokoKeri'
+  physicsImpulseX: 40000
   sprite: s32'zmb204_06', 2
   HairReset: 2
   physicsImpulseX: 80000
@@ -2644,8 +2644,8 @@
   superArmorHitstop: 0, 0
   sprite: s32'zmb232_02', 2
   AttackVoice: 2, -1
-  sprite: s32'zmb232_03', 2
-  sprite: s32'zmb232_04', 2
+  sprite: s32'zmb232_03', 1
+  sprite: s32'zmb232_04', 1
   characterSfx: s32'ARC_BTL_ZMB_Atk2C_Bishi'
   sprite: s32'zmb232_05', 3
   characterSfx: s32'ARC_BTL_ZMB_Atk2C_Attack'
@@ -3707,9 +3707,9 @@
   sprite: s32'zmb402_01', 2
   sprite: s32'zmb402_02', 2
   ForceRandomVoice: s16'vxxx204', 100, s16'vxxx206', 100, s16'', 0, s16'', 0
-  sprite: s32'zmb402_03', 2
-  sprite: s32'zmb402_04', 2
-  sprite: s32'zmb402_05', 2
+  sprite: s32'zmb402_03', 1
+  sprite: s32'zmb402_04', 1
+  sprite: s32'zmb402_05', 1
   sprite: s32'zmb402_06', 1
   sprite: s32'zmb402_07', 1
   sprite: s32'zmb402_08', 2
@@ -3765,18 +3765,18 @@
   FaceReset: 20
   HairReset: 1
   characterSfx: s32'ARC_BTL_ZMB_Hkrnkb_Syubisi'
-  sprite: s32'zmb402_01', 2
-  sprite: s32'zmb402_02', 2
+  sprite: s32'zmb402_01', 1
+  sprite: s32'zmb402_02', 1
   ForceRandomVoice: s16'vxxx204', 100, s16'vxxx206', 100, s16'', 0, s16'', 0
   sprite: s32'zmb402_03', 1
   sprite: s32'zmb402_04', 1
   sprite: s32'zmb402_05', 1
   sprite: s32'zmb402_06', 1
   sprite: s32'zmb402_07', 1
-  sprite: s32'zmb402_08', 3
-  sprite: s32'zmb402_09', 3
-  sprite: s32'zmb402_10', 3
-  sprite: s32'zmb402_11', 2
+  sprite: s32'zmb402_08', 2
+  sprite: s32'zmb402_09', 1
+  sprite: s32'zmb402_10', 1
+  sprite: s32'zmb402_11', 1
   sprite: s32'zmb402_12', 3
   HairReset: 4
   createObject: s32'Hikarinokabe_Tame', (PRIVATE_0)
@@ -3806,7 +3806,7 @@
   endLabel: 
   exitState: 
   beginLabel: s32'hikou'
-  sprite: s32'zmb402_22', 3
+  sprite: s32'zmb402_22', 2
   clearRegisteredUponCode: (FRAME_STEP)
   sprite: s32'zmb402_23', 2
   sprite: s32'zmb402_24', 2
@@ -3829,16 +3829,16 @@
   FaceReset: 20
   HairReset: 1
   characterSfx: s32'ARC_BTL_ZMB_Hkrnkb_Syubisi'
-  sprite: s32'zmb402_01', 2
+  sprite: s32'zmb402_01', 1
   ForceRandomVoice: s16'vxxx204', 100, s16'vxxx206', 100, s16'', 0, s16'', 0
-  sprite: s32'zmb402_02', 2
-  sprite: s32'zmb402_03', 2
-  sprite: s32'zmb402_04', 2
-  sprite: s32'zmb402_05', 2
+  sprite: s32'zmb402_02', 1
+  sprite: s32'zmb402_03', 1
+  sprite: s32'zmb402_04', 1
+  sprite: s32'zmb402_05', 1
   sprite: s32'zmb402_06', 1
   sprite: s32'zmb402_07', 1
-  sprite: s32'zmb402_08', 2
-  sprite: s32'zmb402_11', 2
+  sprite: s32'zmb402_08', 1
+  sprite: s32'zmb402_11', 1
   sprite: s32'zmb402_12', 3
   HairReset: 4
   createObject: s32'Hikarinokabe_Tame', (PRIVATE_0)
@@ -3899,9 +3899,9 @@
   sprite: s32'zmb402_19', 2
   sprite: s32'zmb402_02ex01', 2
   ForceRandomVoice: s16'vxxx204', 100, s16'vxxx206', 100, s16'', 0, s16'', 0
-  sprite: s32'zmb402_03ex01', 2
-  sprite: s32'zmb402_04ex01', 2
-  sprite: s32'zmb402_05ex01', 2
+  sprite: s32'zmb402_03ex01', 1
+  sprite: s32'zmb402_04ex01', 1
+  sprite: s32'zmb402_05ex01', 1
   sprite: s32'zmb402_06ex01', 1
   sprite: s32'zmb402_07ex01', 1
   sprite: s32'zmb402_08ex01', 2
@@ -3948,7 +3948,7 @@
   endLabel: 
   exitState: 
   beginLabel: s32'hikou'
-  sprite: s32'zmb402_22', 3
+  sprite: s32'zmb402_22', 2
   clearRegisteredUponCode: (FRAME_STEP)
   sprite: s32'zmb402_23', 2
   sprite: s32'zmb402_24', 2
@@ -3968,25 +3968,25 @@
       storeValue: (VARIABLE), 48, (STATIC), 1
       endIf: 
     endUpon: 
-  sprite: s32'zmb402_18', 2
+  sprite: s32'zmb402_18', 1
   FaceReset: 20
   HairReset: 1
   PushSpeedX: 
   PushSpeedY: 
   haltMomentum: 1
   characterSfx: s32'ARC_BTL_ZMB_Hkrnkb_Syubisi'
-  sprite: s32'zmb402_19', 2
-  sprite: s32'zmb402_02ex01', 2
-  ForceRandomVoice: s16'vxxx204', 100, s16'vxxx206', 100, s16'', 0, s16'', 0
+  sprite: s32'zmb402_19', 1
+  sprite: s32'zmb402_02ex01', 1
   sprite: s32'zmb402_03ex01', 1
+  ForceRandomVoice: s16'vxxx204', 100, s16'vxxx206', 100, s16'', 0, s16'', 0
   sprite: s32'zmb402_04ex01', 1
   sprite: s32'zmb402_05ex01', 1
   sprite: s32'zmb402_06ex01', 1
   sprite: s32'zmb402_07ex01', 1
-  sprite: s32'zmb402_08ex01', 3
-  sprite: s32'zmb402_09ex01', 3
-  sprite: s32'zmb402_10ex01', 3
-  sprite: s32'zmb402_11ex01', 2
+  sprite: s32'zmb402_08ex01', 2
+  sprite: s32'zmb402_09ex01', 1
+  sprite: s32'zmb402_10ex01', 1
+  sprite: s32'zmb402_11ex01', 1
   sprite: s32'zmb402_12ex01', 3
   HairReset: 4
   createObject: s32'Hikarinokabe_Tame', (PRIVATE_0)
@@ -4029,7 +4029,7 @@
   endLabel: 
   exitState: 
   beginLabel: s32'hikou'
-  sprite: s32'zmb402_22', 3
+  sprite: s32'zmb402_22', 2
   clearRegisteredUponCode: (FRAME_STEP)
   sprite: s32'zmb402_23', 2
   sprite: s32'zmb402_24', 2
@@ -4057,16 +4057,16 @@
   PushSpeedY: 
   haltMomentum: 1
   characterSfx: s32'ARC_BTL_ZMB_Hkrnkb_Syubisi'
-  sprite: s32'zmb402_19', 2
-  sprite: s32'zmb402_02ex01', 2
+  sprite: s32'zmb402_19', 1
+  sprite: s32'zmb402_02ex01', 1
   ForceRandomVoice: s16'vxxx204', 100, s16'vxxx206', 100, s16'', 0, s16'', 0
-  sprite: s32'zmb402_03ex01', 2
-  sprite: s32'zmb402_04ex01', 2
-  sprite: s32'zmb402_05ex01', 2
+  sprite: s32'zmb402_03ex01', 1
+  sprite: s32'zmb402_04ex01', 1
+  sprite: s32'zmb402_05ex01', 1
   sprite: s32'zmb402_06ex01', 1
   sprite: s32'zmb402_07ex01', 1
-  sprite: s32'zmb402_08ex01', 2
-  sprite: s32'zmb402_11ex01', 2
+  sprite: s32'zmb402_08ex01', 1
+  sprite: s32'zmb402_11ex01', 1
   sprite: s32'zmb402_12ex01', 3
   HairReset: 4
   createObject: s32'Hikarinokabe_Tame', (PRIVATE_0)
@@ -4109,7 +4109,7 @@
   endLabel: 
   exitState: 
   beginLabel: s32'hikou'
-  sprite: s32'zmb402_22', 3
+  sprite: s32'zmb402_22', 2
   clearRegisteredUponCode: (FRAME_STEP)
   sprite: s32'zmb402_23', 2
   sprite: s32'zmb402_24', 2
@@ -4187,8 +4187,8 @@
   sprite: s32'zmb404_00', 2
   FaceReset: 20
   HairReset: 1
-  sprite: s32'zmb404_01', 2
-  sprite: s32'zmb404_02', 2
+  sprite: s32'zmb404_01', 1
+  sprite: s32'zmb404_02', 1
   RandomVoice: s16'vxxx202', 100, s16'vxxx203', 100, s16'vxxx205', 100, s16'', 0
   PrivateBankSE: 0, s32'ARC_BTL_ZMB_Titujo_Chrg', 100
   sprite: s32'zmb404_03', 2
@@ -4205,7 +4205,8 @@
     endUpon: 
   sprite: s32'zmb404_04', 2
   sprite: s32'zmb404_05', 2
-  sprite: s32'zmb404_03', 1
+  sprite: s32'zmb404_03', 2
+  sprite: s32'zmb404_04', 1
   storeValue: (VARIABLE), 46, (STATIC), 1
   clearRegisteredUponCode: (FRAME_STEP)
   cameraShake: s32'ShakeTate', 500, 0, 30, 15
