--- 1.30/BBS_VGSEF.bbs	2022-06-29 16:26:28.000000000 +0200
+++ 1.31/BBS_VGSEF.bbs	2022-06-29 17:54:22.000000000 +0200
@@ -1,1916 +1,1916 @@
 beginState: s32'dmy'
   sprite: s32'null', 1
   endState: 
 
 beginState: s32'okasi'
   upon: (IMMEDIATE)
     callSubroutine: s32'cmn_okasi'
     endUpon: 
   sprite: s32'null', 2147483647
   linkParticle: s32'vgs_okasi'
   endState: 
 
 beginState: s32'kidan_5D'
   upon: (IMMEDIATE)
     normalAttack: 
     callSubroutine: s32'cmnAtkLevel_2_AtkInit'
     callSubroutine: s32'cmn5DKidan_AtkInit'
     callSubroutine: s32'cmnKidan_Init'
     damage1: 250
     AtkDamageEffect: 4, s32'cmn_BomHit'
     horizontalMomentum: 102
     linkParticle: s32'cmn_kidan'
     upon: 49
       CheckSignalName: s32'Kidan_5D'
       if: (VARIABLE), (Result)
         clearRegisteredUponCode: 49
         clearRegisteredUponCode: (FRAME_STEP)
         upon: (HIT_OR_GUARD_ANY)
           createObject: s32'cmn_kidan_end', (PLAYER)
           endUpon: 
         endIf: 
       endUpon: 
     checkCurrentStateName: 3, s32'NmlAtk5D'
     if: (VARIABLE), (Result)
       copyFromObjectVar: (SELF), (VARIABLE), (Result), (PLAYER), (VARIABLE), 45
       ifNot: (VARIABLE), (Result)
         groundHitEffect: (AIR_FACE_UP)
         airHitEffect: (AIR_FACE_UP)
         hitAirPushbackX: 20000
         hitAirPushbackY: 20000
         endIf: 
       endIf: 
     endUpon: 
   sprite: s32'kidan', 1
   hit: 
   multiplyAlphaColor: 0
   endLabel: 
   sprite: s32'kidan', 2147483647
   multiplyAlphaColor: 255
   physicsImpulseX: 70000
   WorldOutToEliminate: 1
   beginLabel: s32'Koware'
   sprite: s32'null', 10
   haltMomentum: 1
   linkParticle: s32''
   endState: 
 
 beginState: s32'kidan_smoke'
   upon: (IMMEDIATE)
     modifyPositionX: -260000
     setPositionY: 0
     endUpon: 
   sprite: s32'null', 1
   createParticle: s32'bg_undersmoke00', (PLAYER)
   endState: 
 
 beginState: s32'kidan_start'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     createObject: s32'kidan_start_aura', (PLAYER)
     endUpon: 
   sprite: s32'kidan', 3
   setDepth: 40000
   sprite: s32'keep', 1
   setDepth: -60000
   modifyPositionX: -10000
   modifyPositionY: -10000
   sprite: s32'keep', 2
   cmnSfx: s32'ARC_BTL_CMN_kidan_Chrg'
   sprite: s32'keep', 3
   setDepth: -120000
   modifyPositionX: -60000
   modifyPositionY: 30000
   sprite: s32'keep', 3
   setDepth: -100000
   modifyPositionX: 50000
   modifyPositionY: -10000
   endLabel: 
   endState: 
 
 beginState: s32'kidan_start_aura'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     linkParticle: s32'cmn_kidan_throw'
     endUpon: 
   sprite: s32'null', 9
   setObjectSize: 150
   SetScaleSpeed: 50
   sprite: s32'null', 2147483647
   setObjectSize: 600
   SetScaleSpeed: 0
   endState: 
 
 beginState: s32'kidan_start2'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     modifyPositionX: 40000
     createObject: s32'kidan_start2_aura', (PLAYER)
     endUpon: 
   sprite: s32'null', 3
   setDepth: 30000
   sprite: s32'null', 3
   setDepth: 40000
   modifyPositionX: 10000
   modifyPositionY: 10000
   endLabel: 
   endState: 
 
 beginState: s32'kidan_start2_aura'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     linkParticle: s32'cmn_kidan_throw'
     endUpon: 
   sprite: s32'null', 3
   setObjectSize: 300
   SetScaleSpeed: 66
   sprite: s32'null', 2147483647
   setObjectSize: 500
   SetScaleSpeed: 0
   endState: 
 
 beginState: s32'kidan_start3'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     modifyPositionX: 40000
     createObject: s32'kidan_start3_aura', (PLAYER)
     endUpon: 
   sprite: s32'kidan', 3
   setDepth: -90000
   sprite: s32'keep', 3
   setDepth: -100000
   modifyPositionX: 40000
   modifyPositionY: -10000
   endLabel: 
   endState: 
 
 beginState: s32'kidan_start3_aura'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     linkParticle: s32'cmn_kidan_throw'
     endUpon: 
   sprite: s32'null', 3
   setObjectSize: 300
   SetScaleSpeed: 66
   sprite: s32'null', 2147483647
   setObjectSize: 500
   SetScaleSpeed: 0
   endState: 
 
 beginState: s32'Multikidan'
   upon: (IMMEDIATE)
     specialAttack: 
     callSubroutine: s32'cmnAtkLevel_2_AtkInit'
     callSubroutine: s32'cmnSpecialShot_AtkInit'
     setProjectileDurability: 1
     damage1: 100
     tensionGainOnConnect: 200
     untechAmount: 30
     hitStop: 5
     enemyHitstopAddition: -1, 0, 0
     airHitEffect: (AIR_FACE_DOWN)
     hitAirPushbackX: 12500
     hitAirPushbackY: 0
     Unknown778: 4000
     slidingKnockdownDuration: 10
     chipDamagePercent: 30
     AtkDamageEffect: 4, s32'cmn_BomHit'
     AtkShotIgnoreZReversal: 0
     AtkShakeCamera: s32'ShakeYoko', 600, 1, 15, 0
     horizontalMomentum: 102
     LandingHeight: 0
     AutoHitSignalSending: 1
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     SetActionTimeMax: 300
     initializeDestroy: 
     numberOfHits: 1
     KowareCountDamageCollision: 1
     destroyOnEnemyDamage: 1
     destroyOnEnemyGuard: 1
     destroyOnHit: 1
     KowareCountSousai: 1
     upon: (ON_DESTROY)
       deactivateObjectById: (SELF)
       endUpon: 
     upon: (TOUCH_GROUND)
       createParticle: s32'bg_groundsmokeS', (PLAYER_LAND)
       EffectForLandBreak: '0x00000000E80300000000000000000000'
       deactivateObjectById: (SELF)
       endUpon: 
     upon: (HIT_OR_GUARD_ANY)
       createObject: s32'cmn_kidan_end', (PLAYER)
       deactivateObjectById: (SELF)
       endUpon: 
     upon: 87
       clearRegisteredUponCode: 87
       createObject: s32'cmn_reversalkidan', (PLAYER)
       deactivateObjectById: (SELF)
       linkParticle: s32''
       endUpon: 
     physicsImpulseX: 90000
     modifyPositionY: 40000
     upon: (ON_HIT_OR_BLOCK)
       modifyPositionY: -60000
       hitAirPushbackY: 24000
       slidingKnockdownDuration: 0
       airHitEffect: (AIR_FACE_UP)
       AtkGravityReset: 
       endUpon: 
     upon: (CLASH)
       modifyPositionY: 10000
       untechAmount: 26
       hitAirPushbackY: 24000
       groundHitEffect: (AIR_FACE_UP)
       airHitEffect: (AIR_FACE_UP)
       AtkGravityReset: 
       endUpon: 
     damageSoundEffect: s32'ARC_BTL_CMN_Hit_Kidan'
     guardSoundEffect: s32'ARC_BTL_CMN_Guard_Kidan'
     upon: (FRAME_STEP)
       endUpon: 
     linkParticle: s32'cmn_kidan'
     endUpon: 
   sprite: s32'kidan', 2147483647
   hit: 
   endState: 
 
 beginState: s32'vgs_Multikidan_Mazzle'
   sprite: s32'null', 1
   createParticle: s32'vgs_Multikidan_Mazzle', (PLAYER)
   endState: 
 
 beginState: s32'Multikidan_start'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     createObject: s32'Multikidan_start_aura', (PLAYER)
     endUpon: 
   sprite: s32'null', 3
   setDepth: 40000
-  sprite: s32'null', 3
+  sprite: s32'null', 2
   setDepth: -60000
   modifyPositionX: -10000
   modifyPositionY: -10000
   sprite: s32'null', 2
   setDepth: -120000
   modifyPositionX: -60000
   modifyPositionY: 30000
-  sprite: s32'null', 2
+  sprite: s32'null', 1
   setDepth: -100000
   modifyPositionX: 50000
   modifyPositionY: -10000
   endLabel: 
   endState: 
 
 beginState: s32'Multikidan_start_aura'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     linkParticle: s32'cmn_kidan_throw'
     endUpon: 
   sprite: s32'null', 24
   setObjectSize: 150
   SetScaleSpeed: 18
   sprite: s32'null', 2147483647
   setObjectSize: 600
   SetScaleSpeed: 0
   endState: 
 
 beginState: s32'Multikidan_start2'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     modifyPositionX: 40000
     createObject: s32'Multikidan_start2_aura', (PLAYER)
     endUpon: 
   sprite: s32'kidan', 1
   setDepth: 30000
   sprite: s32'keep', 1
   setDepth: 40000
   modifyPositionX: 10000
   modifyPositionY: 10000
   endLabel: 
   endState: 
 
 beginState: s32'Multikidan_start2_aura'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     linkParticle: s32'cmn_kidan_throw'
     endUpon: 
   sprite: s32'null', 2
   setObjectSize: 300
   SetScaleSpeed: 100
   sprite: s32'null', 2147483647
   setObjectSize: 500
   SetScaleSpeed: 0
   endState: 
 
 beginState: s32'Multikidan_start3'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     modifyPositionX: 40000
     createObject: s32'Multikidan_start3_aura', (PLAYER)
     endUpon: 
   sprite: s32'null', 1
   setDepth: -90000
   sprite: s32'null', 1
   setDepth: -100000
   modifyPositionX: 40000
   modifyPositionY: -10000
   endLabel: 
   endState: 
 
 beginState: s32'Multikidan_start3_aura'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     linkParticle: s32'cmn_kidan_throw'
     endUpon: 
   sprite: s32'null', 2
   setObjectSize: 300
   SetScaleSpeed: 100
   sprite: s32'null', 2147483647
   setObjectSize: 500
   SetScaleSpeed: 0
   endState: 
 
 beginState: s32'kidan_Air'
   upon: (IMMEDIATE)
     normalAttack: 
     callSubroutine: s32'cmnAtkLevel_2_AtkInit'
     callSubroutine: s32'cmnAir5DKidan_AtkInit'
     callSubroutine: s32'cmnKidan_Init'
     damage1: 250
     hitAirPushbackX: 5000
     hitAirPushbackY: 5000
     horizontalMomentum: 102
     verticalMomentum: 102
     LandingHeight: 0
     AtkDamageEffect: 4, s32'cmn_BomHit'
     upon: 49
       CheckSignalName: s32'kidan_Air'
       if: (VARIABLE), (Result)
         clearRegisteredUponCode: 49
         clearRegisteredUponCode: (FRAME_STEP)
         upon: (HIT_OR_GUARD_ANY)
           createObject: s32'cmn_kidan_end', (PLAYER)
           endUpon: 
         endIf: 
       endUpon: 
     checkCurrentStateName: 3, s32'NmlAtkAir5D'
     if: (VARIABLE), (Result)
       copyFromObjectVar: (SELF), (VARIABLE), (Result), (PLAYER), (VARIABLE), 45
       ifNot: (VARIABLE), (Result)
         groundHitEffect: (AIR_FACE_UP)
         airHitEffect: (AIR_FACE_UP)
         hitAirPushbackX: 20000
         hitAirPushbackY: 20000
         endIf: 
       endIf: 
     upon: (ON_HIT_OR_BLOCK)
       physicsImpulseY: -27000
       endUpon: 
     upon: (CLASH)
       physicsImpulseY: -30000
       endUpon: 
     linkParticle: s32'cmn_kidan'
     endUpon: 
   sprite: s32'kidan_nanamesita0', 1
   hit: 
   displayModel: 0
   sprite: s32'kidan_nanamesita', 100
   displayModel: 1
   setAngle: -30000
   physicsImpulseX: 60000
   beginLabel: s32'Koware'
   sprite: s32'null', 10
   haltMomentum: 1
   linkParticle: s32''
   endState: 
 
 beginState: s32'jumpkidan_start'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 3
     createObject: s32'jumpkidan_start_aura', (PLAYER)
     endUpon: 
   sprite: s32'null', 2
   setDepth: -80000
   sprite: s32'null', 2
   setDepth: -110000
   modifyPositionX: -50000
   modifyPositionY: 50000
   sprite: s32'null', 2
   cmnSfx: s32'ARC_BTL_CMN_kidan_Chrg'
   sprite: s32'null', 2
   setDepth: -110000
   modifyPositionX: 10000
   modifyPositionY: 50000
   sprite: s32'null', 3
   setDepth: 80000
   modifyPositionX: 130000
   modifyPositionY: -180000
   endLabel: 
   endState: 
 
 beginState: s32'jumpkidan_start_aura'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     linkParticle: s32'cmn_kidan_throw'
     endUpon: 
   sprite: s32'null', 8
   setObjectSize: 350
   SetScaleSpeed: 50
   sprite: s32'null', 2147483647
   setObjectSize: 750
   SetScaleSpeed: 0
   endState: 
 
 beginState: s32'jumpkidan2_start'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 3
     createObject: s32'jumpkidan2_start_aura', (PLAYER)
     endUpon: 
   sprite: s32'null', 2
   setDepth: 80000
   sprite: s32'null', 2
   setDepth: 110000
   modifyPositionX: 60000
   modifyPositionY: -10000
   sprite: s32'null', 2
   setDepth: 60000
   modifyPositionX: 140000
   modifyPositionY: -120000
   endState: 
 
 beginState: s32'jumpkidan2_start_aura'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     linkParticle: s32'cmn_kidan_throw'
     endUpon: 
   sprite: s32'null', 4
   setObjectSize: 400
   SetScaleSpeed: 87
   sprite: s32'null', 2147483647
   setObjectSize: 750
   SetScaleSpeed: 0
   endState: 
 
 beginState: s32'jumpkidan3_start'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 3
     createObject: s32'jumpkidan3_start_aura', (PLAYER)
     endUpon: 
   sprite: s32'null', 2
   setDepth: -160000
   sprite: s32'null', 2
   setDepth: -90000
   modifyPositionX: 50000
   modifyPositionY: -110000
   sprite: s32'null', 2
   setDepth: 90000
   modifyPositionX: 70000
   modifyPositionY: -70000
   endState: 
 
 beginState: s32'jumpkidan3_start_aura'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     linkParticle: s32'cmn_kidan_throw'
     endUpon: 
   sprite: s32'null', 4
   setObjectSize: 300
   SetScaleSpeed: 112
   sprite: s32'null', 2147483647
   setObjectSize: 750
   SetScaleSpeed: 0
   endState: 
 
 beginState: s32'Multikidan_Air'
   upon: (IMMEDIATE)
     specialAttack: 
     callSubroutine: s32'cmnAtkLevel_2_AtkInit'
     callSubroutine: s32'cmnSpecialShot_AtkInit'
     setProjectileDurability: 1
     damage1: 100
     tensionGainOnConnect: 200
     callSubroutine: s32'cmn_AssistShotHosei'
     hitStop: 7
     enemyHitstopAddition: -3, 0, 0
     groundHitEffect: (AIR_FACE_UP)
     airHitEffect: (AIR_FACE_UP)
     hitAirPushbackX: 15000
     hitAirPushbackY: 20000
     chipDamagePercent: 30
     AtkDamageEffect: 4, s32'cmn_BomHit'
     AtkShotIgnoreZReversal: 0
     AtkShakeCamera: s32'ShakeYoko', 600, 1, 15, 0
     checkCurrentStateName: 3, s32'AssistAttack'
     if: (VARIABLE), (Result)
       damage1: 130
       AtkHoseiGuardBalance: 1000
       untechAmount: 20
       enemyHitstopAddition: -3, 2, 2
       chipDamagePercent: 0
       endIf: 
     horizontalMomentum: 102
     verticalMomentum: 102
     LandingHeight: 0
     AutoHitSignalSending: 1
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     SetActionTimeMax: 300
     initializeDestroy: 
     numberOfHits: 1
     KowareCountDamageCollision: 1
     destroyOnEnemyDamage: 1
     destroyOnEnemyGuard: 1
     destroyOnHit: 1
     KowareCountSousai: 1
     upon: (ON_DESTROY)
       deactivateObjectById: (SELF)
       endUpon: 
     upon: (TOUCH_GROUND)
       clearRegisteredUponCode: (TOUCH_GROUND)
       createParticle: s32'bg_groundsmokeS', (PLAYER_LAND)
       EffectForLandBreak: '0x00000000E80300000000000000000000'
       jumpToLabel: s32'Koware'
       endUpon: 
     upon: (HIT_OR_GUARD_ANY)
       createObject: s32'cmn_kidan_end', (PLAYER)
       deactivateObjectById: (SELF)
       endUpon: 
     upon: 87
       clearRegisteredUponCode: 87
       createObject: s32'cmn_reversalkidan', (PLAYER)
       deactivateObjectById: (SELF)
       linkParticle: s32''
       endUpon: 
     upon: (ON_HIT_OR_BLOCK)
       physicsImpulseY: -45000
       endUpon: 
     upon: (CLASH)
       physicsImpulseY: -50000
       endUpon: 
     damageSoundEffect: s32'ARC_BTL_CMN_Hit_Kidan'
     guardSoundEffect: s32'ARC_BTL_CMN_Guard_Kidan'
     upon: (FRAME_STEP)
       endUpon: 
     linkParticle: s32'cmn_kidan'
     endUpon: 
   sprite: s32'multi_kidan_nanamesita0', 1
   hit: 
   displayModel: 0
   sprite: s32'multi_kidan_nanamesita', 100
   displayModel: 1
   setAngle: -30000
   physicsImpulseX: 80000
   beginLabel: s32'Koware'
   sprite: s32'null', 5
   linkParticle: s32''
   haltMomentum: 1
   endState: 
 
 beginState: s32'vgs_AirMultikidan_Mazzle'
   upon: (IMMEDIATE)
     LinkObjectPosition: 0
     setAngle: -30000
     endUpon: 
   sprite: s32'null', 100
   linkParticle: s32'vgs_Multikidan_Mazzle'
   endState: 
 
 beginState: s32'AssistMultikidan_start'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 3
     createObject: s32'jumpkidan_start_aura', (PLAYER)
     endUpon: 
   sprite: s32'null', 3
   setDepth: -80000
   sprite: s32'null', 3
   setDepth: -110000
   modifyPositionX: -50000
   modifyPositionY: 50000
   sprite: s32'null', 3
   sprite: s32'null', 3
   setDepth: -110000
   modifyPositionX: 10000
   modifyPositionY: 50000
   sprite: s32'null', 3
   setDepth: 80000
   modifyPositionX: 130000
   modifyPositionY: -180000
   endLabel: 
   endState: 
 
 beginState: s32'AirMultikidan_start'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 3
     createObject: s32'jumpkidan_start_aura', (PLAYER)
     endUpon: 
   sprite: s32'null', 2
   setDepth: -80000
-  sprite: s32'null', 2
+  sprite: s32'null', 1
   setDepth: -110000
   modifyPositionX: -50000
   modifyPositionY: 50000
   sprite: s32'null', 2
-  sprite: s32'null', 2
+  sprite: s32'null', 1
   setDepth: -110000
   modifyPositionX: 10000
   modifyPositionY: 50000
   sprite: s32'null', 2
   setDepth: 80000
   modifyPositionX: 130000
   modifyPositionY: -180000
   endLabel: 
   endState: 
 
 beginState: s32'AirMultikidan_start_aura'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     linkParticle: s32'cmn_kidan_throw'
     endUpon: 
   sprite: s32'null', 12
   setObjectSize: 350
   SetScaleSpeed: 33
   sprite: s32'null', 2147483647
   setObjectSize: 750
   SetScaleSpeed: 0
   endState: 
 
 beginState: s32'AirMultikidan2_start'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 3
     createObject: s32'AirMultikidan2_start_aura', (PLAYER)
     endUpon: 
   sprite: s32'null', 1
   setDepth: 80000
   sprite: s32'null', 1
   setDepth: 110000
   modifyPositionX: 60000
   modifyPositionY: -10000
   sprite: s32'null', 1
   setDepth: 60000
   modifyPositionX: 140000
   modifyPositionY: -120000
   endState: 
 
 beginState: s32'AirMultikidan2_start_aura'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     linkParticle: s32'cmn_kidan_throw'
     endUpon: 
   sprite: s32'null', 2
   setObjectSize: 400
   SetScaleSpeed: 175
   sprite: s32'null', 2147483647
   setObjectSize: 750
   SetScaleSpeed: 0
   endState: 
 
 beginState: s32'AirMultikidan3_start'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 3
     createObject: s32'AirMultikidan3_start_aura', (PLAYER)
     endUpon: 
   sprite: s32'null', 1
   setDepth: -160000
   sprite: s32'null', 1
   setDepth: -90000
   modifyPositionX: 50000
   modifyPositionY: -110000
   sprite: s32'null', 2
   setDepth: 90000
   modifyPositionX: 70000
   modifyPositionY: -70000
   endState: 
 
 beginState: s32'AirMultikidan3_start_aura'
   upon: (IMMEDIATE)
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     linkParticle: s32'cmn_kidan_throw'
     endUpon: 
   sprite: s32'null', 2
   setObjectSize: 300
   SetScaleSpeed: 225
   sprite: s32'null', 2147483647
   setObjectSize: 750
   SetScaleSpeed: 0
   endState: 
 
 beginState: s32'vgs_hajiki'
   upon: (IMMEDIATE)
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     LinkObjectStop: 2
     endUpon: 
   sprite: s32'null', 30
   linkParticle: s32'vgs_hajiki'
   endState: 
 
 beginState: s32'VGS_202Eff'
   upon: (IMMEDIATE)
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     modifyPositionY: 62000
     modifyPositionX: 62000
     setAngle: 45000
     setObjectSize: 750
     endUpon: 
   sprite: s32'null', 60
   CreateArg_Angle: 0, 45000
   CreateParticleArg: s32'vgs_202Eff00', 100
   createObject: s32'VGS_202Eff_smoke', (PLAYER)
   createObject: s32'VGS_202Eff_smoke2', (PLAYER)
   createObject: s32'VGS_202Eff_smoke3', (PLAYER)
   createObject: s32'VGS_202Eff_smoke4', (PLAYER)
   endState: 
 
 beginState: s32'VGS_202Eff_smoke'
   upon: (IMMEDIATE)
     modifyPositionX: -200000
     setPositionY: 0
     endUpon: 
   sprite: s32'null', 1
   CreateArg_Scale: 0, 750
   CreateParticleArg: s32'bg_undersmoke00', 100
   endState: 
 
 beginState: s32'VGS_202Eff_smoke2'
   upon: (IMMEDIATE)
     modifyPositionX: -200000
     setPositionY: 0
     endUpon: 
   sprite: s32'null', 1
   CreateArg_Scale: 0, 1250
   CreateParticleArg: s32'bg_undersmoke00', 100
   endState: 
 
 beginState: s32'VGS_202Eff_smoke3'
   upon: (IMMEDIATE)
     flipDirection: 
     modifyPositionX: 50000
     setPositionY: 0
     endUpon: 
   sprite: s32'null', 1
   CreateArg_Scale: 0, 400
   CreateParticleArg: s32'bg_undersmoke00', 100
   endState: 
 
 beginState: s32'VGS_202Eff_smoke4'
   upon: (IMMEDIATE)
     flipDirection: 
     modifyPositionX: 150000
     setPositionY: 0
     endUpon: 
   sprite: s32'null', 1
   CreateArg_Scale: 0, 800
   CreateParticleArg: s32'bg_undersmoke00', 100
   endState: 
 
 beginState: s32'VGS_5CEff'
   upon: (IMMEDIATE)
     IgnoreWorldStopByAll: 1
     setAngle: 80000
     modifyPositionY: 75000
     modifyPositionX: 200000
     endUpon: 
   sprite: s32'null', 60
   linkParticle: s32'cmn_strike00'
   endState: 
 
 beginState: s32'VGS_2CEff'
   upon: (IMMEDIATE)
     IgnoreWorldStopByAll: 1
     modifyPositionY: 250000
     setAngle: -8000
     endUpon: 
   sprite: s32'null', 60
   linkParticle: s32'cmn_strike00'
   endState: 
 
 beginState: s32'VGS_Air5CEff'
   upon: (IMMEDIATE)
     IgnoreWorldStopByAll: 1
     endUpon: 
   sprite: s32'null', 60
   linkParticle: s32'cmn_strike00'
   endState: 
 
 beginState: s32'VGS_Air2CEff'
   upon: (IMMEDIATE)
     IgnoreWorldStopByAll: 1
     modifyPositionY: 250000
     setAngle: -10000
     endUpon: 
   sprite: s32'null', 60
   linkParticle: s32'cmn_strike00'
   endState: 
 
 beginState: s32'VGS_400Eff'
   upon: (IMMEDIATE)
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     LinkObjectStop: 2
     modifyPositionY: 220000
     modifyPositionX: 260000
     jumpToStateInObject: (SELF), s32'End'
     endUpon: 
   sprite: s32'null', 2147483647
   linkParticle: s32'vgs_400Eff00'
   createParticle: s32'vgs_400Eff_nokosi', (PLAYER)
   createObject: s32'VGS_400_groundSmoke', (PLAYER)
   beginLabel: s32'End'
   sprite: s32'null', 15
   linkParticle: s32'vgs_400Eff01'
   deactivateObjectByName: s32'VGS_400_groundSmoke'
   LinkObjectPosition: 0
   LinkObjectStop: 0
   endLabel: 
   endState: 
 
 beginState: s32'VGS_400_Assist_Eff'
   upon: (IMMEDIATE)
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     LinkObjectStop: 2
     modifyPositionY: 220000
     modifyPositionX: 260000
     jumpToStateInObject: (SELF), s32'End'
     endUpon: 
   sprite: s32'null', 2147483647
   linkParticle: s32'vgs_400Eff00_Assist'
   createParticle: s32'vgs_400Eff_nokosi_Assist', (PLAYER)
   beginLabel: s32'End'
   sprite: s32'null', 15
   linkParticle: s32'vgs_400Eff01_Assist'
   LinkObjectPosition: 0
   LinkObjectStop: 0
   endLabel: 
   endState: 
 
 beginState: s32'VGS_400_groundSmoke'
   upon: (IMMEDIATE)
     setPositionY: 0
     LinkObjectStop: 2
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     modifyPositionX: -75000
     endUpon: 
   beginLabel: s32'loop'
   sprite: s32'null', 3
   CreateParticleArg: s32'bg_undersmoke00', 100
   endLabel: 
   gotoLabel: s32'loop'
   endState: 
 
 beginState: s32'VGS_400EffAir'
   upon: (IMMEDIATE)
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     LinkObjectStop: 2
     modifyPositionY: 220000
     modifyPositionX: 260000
     jumpToStateInObject: (SELF), s32'End'
     endUpon: 
   sprite: s32'null', 2147483647
   linkParticle: s32'vgs_400Eff00'
   createParticle: s32'vgs_400Eff_nokosi', (PLAYER)
   beginLabel: s32'End'
   sprite: s32'null', 15
   linkParticle: s32'vgs_400Eff01'
   LinkObjectPosition: 0
   LinkObjectStop: 0
   endLabel: 
   endState: 
 
 beginState: s32'VGS_400B1Eff'
   upon: (IMMEDIATE)
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     LinkObjectStop: 2
     jumpToStateInObject: (SELF), s32'End'
     endUpon: 
   sprite: s32'null', 2147483647
   linkParticle: s32'vgs_400Effb1_00'
   beginLabel: s32'End'
   sprite: s32'null', 15
   linkParticle: s32'vgs_400Effb1_01'
   LinkObjectPosition: 0
   LinkObjectStop: 0
   endLabel: 
   endState: 
 
 beginState: s32'VGS_400B2Eff'
   upon: (IMMEDIATE)
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     LinkObjectStop: 2
     jumpToStateInObject: (SELF), s32'End'
     endUpon: 
   sprite: s32'null', 2147483647
   linkParticle: s32'vgs_400Effb2_00'
   beginLabel: s32'End'
   sprite: s32'null', 15
   linkParticle: s32'vgs_400Effb2_01'
   LinkObjectStop: 0
   endLabel: 
   endState: 
 
 beginState: s32'VGS_402Eff'
   upon: (IMMEDIATE)
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     modifyPositionX: 62500
     modifyPositionY: 125000
     endUpon: 
   sprite: s32'null', 2147483647
   linkParticle: s32'vgs_402_speedEff'
   endState: 
 
 beginState: s32'BigbangLand'
   upon: (IMMEDIATE)
     overdriveAttack: 
     callSubroutine: s32'cmnAtkLevel_4_AtkInit'
     callSubroutine: s32'cmnUltimateShot_AtkInit'
     damage1: 2200
     minimalDamagePercent: 37
     hitStop: 15
     groundHitEffect: 12
     airHitEffect: 12
     hitAirPushbackX: 48000
     hitAirPushbackY: 40000
     untechAmount: 40
     AtkFrontDirection: 0
     AtkDamageEffect: 1, s32' '
     AtkShakeCamera: s32'ShakeTateYoko', 1000, 0, 45, 10
     disableHitstopForSelf: 0
     AutoHitStopSending: 1
     AutoHitSignalSending: 1
     physicsImpulseX: 100000
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     SetActionTimeMax: 120
     upon: (AFTER_EXIT)
       clearRegisteredUponCode: (AFTER_EXIT)
       storeValue: (VARIABLE), 46, (STATIC), 1
       endUpon: 
     upon: (ON_DESTROY)
       deactivateObjectById: (SELF)
       ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (PosY), (STATIC), 500000
         EffectForLandBreak: '0x01000000E80300000000000000000000'
         endIf: 
       createObject: s32'Bigbang_end', (PLAYER)
       ifNot: (VARIABLE), 46
         checkCurrentStateName: 3, s32'BigBangAttack'
         if: (VARIABLE), (Result)
           SendSignal: 3, 23
           endIf: 
         checkCurrentStateName: 3, s32'BigBangAttack_Air_Yoko'
         if: (VARIABLE), (Result)
           SendSignal: 3, 23
           endIf: 
         endIf: 
       endUpon: 
     upon: (ON_ATTACK_COLLISION_MAIN_PLAYER)
       clearRegisteredUponCode: (ON_ATTACK_COLLISION_MAIN_PLAYER)
       requestDestroy: (SELF)
       endUpon: 
     upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE)
       clearRegisteredUponCode: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE)
       callSubroutine: s32'cmnUltimate_CameraCombo'
       damage1: 2200
       endUpon: 
     upon: (ON_ATTACK_COLLISION_SOUSAI)
       ifNot: (VARIABLE), 45
         PushSpeedX: 
         PushSpeedY: 
         endIf: 
       storeValue: (VARIABLE), 47, (STATIC), 0
       upon: 77
         ifNot: (VARIABLE), (CheckStop)
           modifyVar: (ADD), (VARIABLE), 47, (STATIC), 1
           ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 47, (STATIC), 5
             clearRegisteredUponCode: 77
             haltMomentum: 1
             PopSpeedX: 
             PopSpeedY: 
             endIf: 
           endIf: 
         endUpon: 
       damage1: 200
       physicsImpulseX: 30000
       modifyPositionX: -150000
       horizontalMomentum: 110
       hit: 
       modifyVar: (ADD), (VARIABLE), 45, (STATIC), 1
       ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 45, (STATIC), 8
         requestDestroy: (SELF)
         ifNot: (VARIABLE), 46
           checkCurrentStateName: 3, s32'BigBangAttack'
           if: (VARIABLE), (Result)
             SendSignal: 3, 23
             endIf: 
           checkCurrentStateName: 3, s32'BigBangAttack_Air_Yoko'
           if: (VARIABLE), (Result)
             SendSignal: 3, 23
             endIf: 
           endIf: 
         endIf: 
       else: 
         ifNot: (VARIABLE), 46
           checkCurrentStateName: 3, s32'BigBangAttack'
           if: (VARIABLE), (Result)
             SendSignal: 3, 25
             endIf: 
           checkCurrentStateName: 3, s32'BigBangAttack_Air_Yoko'
           if: (VARIABLE), (Result)
             SendSignal: 3, 25
             endIf: 
           endIf: 
         endElse: 
       endUpon: 
     upon: (FRAME_STEP)
       ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (PosY), (STATIC), 0
         createParticle: s32'bg_groundsmokeL', (PLAYER)
         EffectForLandBreak: '0x00000000E80300000000000000000000'
         deactivateObjectById: (SELF)
         endIf: 
       IfBeginCoType: 3, 2, 32
         IfBeginCoType: 32, 2, 32
           AtkNoSousai: 1
           endIf: 
         else: 
           AtkNoSousai: 0
           endElse: 
         endIf: 
       IfBeginNotCoType: 3, 2, 32
         IfBeginNotCoType: 32, 2, 32
           AtkNoSousai: 1
           endIf: 
         else: 
           AtkNoSousai: 0
           endElse: 
         endIf: 
       endUpon: 
     endUpon: 
   sprite: s32'bigbangattack', 100
   linkParticle: s32'vgs_430Jizoku'
   createObject: s32'BigBangNokosiEff', (PLAYER)
   hit: 
   ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (PosY), (STATIC), 400000
     createObject: s32'groundSmoke', (PLAYER)
     endIf: 
   endState: 
 
 beginState: s32'Bigbang_end'
   upon: (IMMEDIATE)
     linkParticle: s32'cmn_bomb_sp'
     cameraShake: s32'ShakeTate', 1500, 5, 45, 20
     IgnoreFinishStop: 1
     endUpon: 
   sprite: s32'null', 10
   cmnSfx: s32'ARC_BTL_CMN_Expl_BB'
   AddWindPowerOffset: 0, 0, 0, 1000, 0, 100, 30
   DestructionCollision: 0, 0, 0, 10000
   sprite: s32'null', 120
   IgnoreFinishStop: 0
   endState: 
 
 beginState: s32'BigBangNokosiEff'
   upon: (IMMEDIATE)
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     endUpon: 
   beginLabel: s32'loop'
   sprite: s32'null', 4
   createObject: s32'BigBangNokosi', (PLAYER)
   endLabel: 
   gotoLabel: s32'loop'
   endState: 
 
 beginState: s32'BigBangNokosi'
   upon: (IMMEDIATE)
     linkParticle: s32'vgs_430nokosi'
     endUpon: 
   sprite: s32'null', 30
   endLabel: 
   endState: 
 
 beginState: s32'groundSmoke'
   upon: (IMMEDIATE)
     setPositionY: 0
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     modifyPositionX: 150000
     endUpon: 
   beginLabel: s32'loop'
   sprite: s32'null', 3
   CreateArg_ScaleMult: 0, 1100
   CreateParticleArg: s32'bg_undersmoke00', 100
   endLabel: 
   gotoLabel: s32'loop'
   endState: 
 
 beginState: s32'BigbangTame'
   upon: (IMMEDIATE)
     IgnoreWorldStopByAll: 1
     LinkObjectChangeToEliminate: 2
     linkParticle: s32'vgs_430Tame'
     jumpToStateInObject: (SELF), s32'end'
     endUpon: 
   sprite: s32'null', 2147483647
   endLabel: 
   beginLabel: s32'end'
   sprite: s32'null', 15
   setObjectSize: 600
   modifyPositionX: -100000
   linkParticle: s32'vgs_430TameEnd'
   endLabel: 
   endState: 
 
 beginState: s32'Bigbang_shadow'
   upon: (IMMEDIATE)
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     IgnoreWorldStopByAll: 1
     jumpToStateInObject: (SELF), s32'end'
     endUpon: 
   sprite: s32'null', 2147483647
   linkParticle: s32'cmn_shadowLine_loop'
   endLabel: 
   beginLabel: s32'end'
   sprite: s32'null', 1
   endLabel: 
   endState: 
 
 beginState: s32'BigbangAir'
   upon: (IMMEDIATE)
     overdriveAttack: 
     callSubroutine: s32'cmnAtkLevel_4_AtkInit'
     callSubroutine: s32'cmnUltimateShot_AtkInit'
     damage1: 2200
     minimalDamagePercent: 35
     hitStop: 15
     groundHitEffect: (AIR_STRONG)
     airHitEffect: (AIR_STRONG)
     hitAirPushbackX: 35000
     hitAirPushbackY: -40000
     untechAmount: 45
     wallBounceCount: 0
     groundBounceCount: 1
     groundBounceYVelocity: 15000
     groundBounceGravity: 20000
     groundBounceYVelocityPercent: 100
     slidingKnockdownDuration: 20
     AtkFrontDirection: 0
     AtkDamageEffect: 1, s32' '
     AtkShakeCamera: s32'ShakeTateYoko', 1000, 0, 45, 10
     disableHitstopForSelf: 0
     AutoHitStopSending: 1
     AutoHitSignalSending: 1
     setAngle: -30000
     SetSpeedByCurrentAngle: 100000, 0
     WorldOutToEliminate: 1
     LinkObjectDamageToEliminate: 0
     SetActionTimeMax: 120
     upon: (AFTER_EXIT)
       clearRegisteredUponCode: (AFTER_EXIT)
       storeValue: (VARIABLE), 46, (STATIC), 1
       endUpon: 
     upon: (ON_DESTROY)
       deactivateObjectById: (SELF)
       ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (PosY), (STATIC), 500000
         EffectForLandBreak: '0x01000000E80300000000000000000000'
         endIf: 
       createObject: s32'Bigbang_end', (PLAYER)
       ifNot: (VARIABLE), 46
         checkCurrentStateName: 3, s32'BigBangAttack_Air_Shita'
         if: (VARIABLE), (Result)
           ifNot: (VARIABLE), 45
             SendSignal: 3, 23
             endIf: 
           endIf: 
         endIf: 
       endUpon: 
     upon: (ON_ATTACK_COLLISION_MAIN_PLAYER)
       clearRegisteredUponCode: (ON_ATTACK_COLLISION_MAIN_PLAYER)
       requestDestroy: (SELF)
       endUpon: 
     upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE)
       clearRegisteredUponCode: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE)
       callSubroutine: s32'cmnUltimate_CameraCombo'
       damage1: 2200
       endUpon: 
     upon: (ON_ATTACK_COLLISION_SOUSAI)
       ifNot: (VARIABLE), 45
         PushSpeedX: 
         PushSpeedY: 
         endIf: 
       storeValue: (VARIABLE), 47, (STATIC), 0
       upon: 77
         ifNot: (VARIABLE), (CheckStop)
           modifyVar: (ADD), (VARIABLE), 47, (STATIC), 1
           ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 47, (STATIC), 5
             clearRegisteredUponCode: 77
             haltMomentum: 1
             PopSpeedX: 
             PopSpeedY: 
             endIf: 
           endIf: 
         endUpon: 
       damage1: 200
       modifyPositionX: -100000
       modifyPositionY: 50000
       SetSpeedByCurrentAngle: 30000, 0
       hit: 
       modifyVar: (ADD), (VARIABLE), 45, (STATIC), 1
       ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 45, (STATIC), 8
         requestDestroy: (SELF)
         ifNot: (VARIABLE), 46
           checkCurrentStateName: 3, s32'BigBangAttack_Air_Shita'
           if: (VARIABLE), (Result)
             SendSignal: 3, 23
             endIf: 
           endIf: 
         endIf: 
       else: 
         ifNot: (VARIABLE), 46
           checkCurrentStateName: 3, s32'BigBangAttack_Air_Shita'
           if: (VARIABLE), (Result)
             SendSignal: 3, 25
             endIf: 
           endIf: 
         endElse: 
       endUpon: 
     upon: (FRAME_STEP)
       ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (PosY), (STATIC), 0
         createParticle: s32'bg_groundsmokeL', (PLAYER)
         EffectForLandBreak: '0x00000000E80300000000000000000000'
         deactivateObjectById: (SELF)
         endIf: 
       IfBeginCoType: 3, 2, 32
         IfBeginCoType: 32, 2, 32
           AtkNoSousai: 1
           endIf: 
         else: 
           AtkNoSousai: 0
           endElse: 
         endIf: 
       IfBeginNotCoType: 3, 2, 32
         IfBeginNotCoType: 32, 2, 32
           AtkNoSousai: 1
           endIf: 
         else: 
           AtkNoSousai: 0
           endElse: 
         endIf: 
       endUpon: 
     endUpon: 
   sprite: s32'bigbangattack', 100
   linkParticle: s32'vgs_430Jizoku'
   createObject: s32'BigBangNokosiEff2', (PLAYER)
   hit: 
   endState: 
 
 beginState: s32'BigBangNokosiEff2'
   upon: (IMMEDIATE)
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     endUpon: 
   beginLabel: s32'loop'
   sprite: s32'null', 4
   createObject: s32'BigBangNokosi2', (PLAYER)
   endLabel: 
   gotoLabel: s32'loop'
   endState: 
 
 beginState: s32'BigBangNokosi2'
   upon: (IMMEDIATE)
     linkParticle: s32'vgs_430nokosi'
     setAngle: -30000
     modifyPositionY: 75000
     endUpon: 
   sprite: s32'null', 30
   endLabel: 
   endState: 
 
 beginState: s32'AirBigbang_shadow'
   upon: (IMMEDIATE)
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     IgnoreWorldStopByAll: 1
     setAngle: -30000
     jumpToStateInObject: (SELF), s32'end'
     endUpon: 
   sprite: s32'null', 2147483647
   linkParticle: s32'cmn_shadowLine_loop'
   endLabel: 
   beginLabel: s32'end'
   sprite: s32'null', 1
   endLabel: 
   endState: 
 
 beginState: s32'FinalFlash_Core'
   upon: (IMMEDIATE)
     LinkObjectStop: 3
     LinkObjectPosition: 3
     SetActionTimeMax: 300
     linkParticle: s32'vgs_432_core00'
     jumpToStateInObject: (SELF), s32'end'
     jumpToStateInObject: 47, s32'end'
     upon: 35
       LinkObjectPosition: 0
       IgnoreWorldStopByAll: 1
       endUpon: 
     upon: (CLASH)
       clearRegisteredUponCode: (CLASH)
       storeValue: (VARIABLE), 45, (STATIC), 0
       triggerUponInObject: s32'FinalFlash_Atk', 29
       checkCurrentStateName: 3, s32'FinalFlash'
       if: (VARIABLE), (Result)
         copyFromObjectVar: (PLAYER), (VARIABLE), 45, (SELF), (STATIC), (Result)
         SendSignal: 3, 25
         endIf: 
       checkCurrentStateName: 3, s32'AirFinalFlash'
       if: (VARIABLE), (Result)
         copyFromObjectVar: (PLAYER), (VARIABLE), 45, (SELF), (STATIC), (Result)
         SendSignal: 3, 25
         endIf: 
       storeValue: (VARIABLE), 47, (STATIC), 1
       storeValue: (VARIABLE), 51, (STATIC), 1
       upon: (CLASH)
         modifyVar: (ADD), (VARIABLE), 47, (STATIC), 1
         ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 47, (STATIC), 16
           checkCurrentStateName: 3, s32'FinalFlash'
           if: (VARIABLE), (Result)
             SendSignal: 3, 24
             endIf: 
           checkCurrentStateName: 3, s32'AirFinalFlash'
           if: (VARIABLE), (Result)
             SendSignal: 3, 24
             endIf: 
           endIf: 
         else: 
           checkCurrentStateName: 3, s32'FinalFlash'
           if: (VARIABLE), (Result)
             SendSignal: 3, 25
             endIf: 
           checkCurrentStateName: 3, s32'AirFinalFlash'
           if: (VARIABLE), (Result)
             SendSignal: 3, 25
             endIf: 
           endElse: 
         endUpon: 
       endUpon: 
     endUpon: 
   beginLabel: s32'loop'
   sprite: s32'null', 2
   modifyVar: (ADD), (VARIABLE), 48, (STATIC), 1
   CreateArg_HikitsugiVal: 2, 48, 2, 51
   createObjectWithArg: s32'FinalFlash_Atk', (PLAYER)
   ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), 45, (STATIC), 9
     applyTo: (PREVIOUS)
       AtkDestructionFinishBeam: 1, 0
       AtkAdditionalHitCount: 1
       endApply: 
     endIf: 
   ifOperation: (IS_GREATER_OR_EQUAL), (VARIABLE), 45, (STATIC), 18
     copyFromObjectVar: (PREVIOUS), (VARIABLE), 45, (SELF), (STATIC), (Angle)
     endIf: 
   sprite: s32'null', 2
   endLabel: 
   modifyVar: (ADD), (VARIABLE), 45, (STATIC), 1
   gotoIfOperation: s32'loop', (IS_LESSER_OR_EQUAL), (VARIABLE), 45, (STATIC), 20
   beginLabel: s32'end'
   sprite: s32'null', 1
   clearRegisteredUponCode: (ON_HIT_OR_BLOCK)
   clearRegisteredUponCode: (CLASH)
   clearRegisteredUponCode: (AFTER_EXIT)
   clearRegisteredUponCode: (FRAME_STEP)
   storeValue: (VARIABLE), 49, (STATIC), 1
   createObject: s32'FinalFlash_Atk', (PLAYER)
   triggerUponInObject: s32'FinalFlash_Atk', 30
   checkCurrentStateName: 3, s32'FinalFlash'
   if: (VARIABLE), (Result)
     SendSignal: 3, 24
     endIf: 
   checkCurrentStateName: 3, s32'AirFinalFlash'
   if: (VARIABLE), (Result)
     SendSignal: 3, 24
     endIf: 
   LinkObjectPosition: 0
   LinkObjectStop: 0
   IgnoreWorldStopByFriend: 1
   sprite: s32'null', 1
   deactivateObjectByName: s32'FinalFlash_Atk'
   sprite: s32'null', 10
   linkParticle: s32'vgs_432_core01'
   sprite: s32'null', 30
   endState: 
 
 beginState: s32'FinalFlash_Atk'
   upon: (IMMEDIATE)
     overdriveAttack: 
     callSubroutine: s32'cmnAtkLevel_4_AtkInit'
     callSubroutine: s32'cmnUltimateShot_AtkInit'
     damage1: 220
     minimalDamagePercent: 38
     AtkHoseiGuardBalance: 500
     hitStop: 1
     groundHitEffect: (AIR_STRONG)
     airHitEffect: (AIR_STRONG)
     hitAirPushbackX: 80000
     hitAirPushbackY: 10000
     pushbackX: 10
     AtkSuikomiY: 1, 30000, 250000, 100, 0
     wallBounceCount: 0
     untechAmount: 150
     AtkRange: 50000000, -320000, 0, 0
     disableOpponentRecovery: 0
     AtkDamageToRecovery: 1
     AtkSousaiSevereCheck: 1
     AtkEffectBeamSousai: 1
     AtkIgnoreSousaiSameLevel: 1
     AtkHitCommonBankSE: 16, s32'ARC_BTL_CMN_Hit_BeamLoop_LP', 0
     AtkGuardCommonBankSE: 17, s32'ARC_BTL_CMN_Grd_BeamLoop_LP', 0
     upon: 30
       groundHitEffect: 12
       airHitEffect: 12
       hitAirPushbackX: 20000
       hitAirPushbackY: 48000
       Unknown778: 2500
       AtkSuikomiY: 0, 0, 0, 0, 0
       AtkNextHitActionName: s32'CmnNeutral'
       IfBeginNotCoType: 3, 2, 24
         hit: 
         endIf: 
       else: 
         recoveryState: 
         endElse: 
       endUpon: 
     LinkObjectStop: 2
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     SetActionTimeMax: 60
     AutoHitSignalSending: 1
     WorldOutToEliminate: 1
     linkParticle: s32'vgs_432_naka00'
     upon: (ON_ATTACK_COLLISION_SOUSAI)
       copyFromObjectVar: (PARENT), (VARIABLE), (ObjectParam1), (SELF), (VARIABLE), (ObjectParam1)
       deactivateObjectById: (SELF)
       SendSignal: 2, 24
       endUpon: 
     upon: 29
       storeValue: (VARIABLE), 45, (STATIC), 0
       endUpon: 
     upon: (ON_ATTACK_HIT_MAIN_PLAYER_DAMAGE)
       callSubroutine: s32'cmnUltimate_CameraCombo'
       if: (VARIABLE), (ObjectParam2)
         checkCurrentStateName: 3, s32'FinalFlash'
         if: (VARIABLE), (Result)
           copyFromObjectVar: (PLAYER), (VARIABLE), 45, (SELF), (STATIC), (Result)
           endIf: 
         checkCurrentStateName: 3, s32'AirFinalFlash'
         if: (VARIABLE), (Result)
           copyFromObjectVar: (PLAYER), (VARIABLE), 45, (SELF), (STATIC), (Result)
           endIf: 
         endIf: 
       endUpon: 
     upon: (HIT_OR_GUARD_ANY)
       if: (VARIABLE), 45
         SendSignal: 2, 23
         endIf: 
       endUpon: 
     upon: (FRAME_STEP)
       IfBeginCoType: 2, 2, 47
         IfBeginNotCoType: 2, 2, 49
           copyFromObjectVar: (SELF), (VARIABLE), (Result), (PARENT), (VARIABLE), (ObjectParam1)
           ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (ObjectParam1), (VARIABLE), 0
             ifNot: (VARIABLE), 47
               deactivateObjectById: (SELF)
               endIf: 
             endIf: 
           endIf: 
         endIf: 
       ifOperation: (IS_LESSER_OR_EQUAL), (VARIABLE), (PosY), (STATIC), 600000
         EffectForLandBreak: '0x03000000E80300000000000000000000'
         endIf: 
       copyFromObjectVar: (SELF), (VARIABLE), (Result), (ENEMY), (VARIABLE), (PlayerDamageOrGuard)
       ifNot: (VARIABLE), (Result)
         StopBankSE: 16, 20
         StopBankSE: 17, 20
         endIf: 
       checkCurrentStateName: 3, s32'FinalFlash'
       ifNot: (VARIABLE), (Result)
         checkCurrentStateName: 3, s32'AirFinalFlash'
         ifNot: (VARIABLE), (Result)
           StopBankSE: 16, 20
           StopBankSE: 17, 20
           endIf: 
         endIf: 
       endUpon: 
     upon: (BEFORE_EXIT)
       CreateArg_ScaleMult: 0, 1000
       CreateArg_AnglePlus: 0, 0
       CreateParticleArg: s32'vgs_432_naka01', 100
       endUpon: 
     endUpon: 
   sprite: s32'finalflash', 1
   hit: 
   modifyPositionX: -100000
   SetSpeedByCurrentAngle: 100000, 0
   sprite: s32'finalflash', 1
   AtkIgnoreSousaiSameLevel: 0
   beginLabel: s32'loop'
   sprite: s32'finalflash', 2
   ScaleForLinkY: 1200
   sprite: s32'finalflash', 2
   ScaleForLinkY: 1000
   endLabel: 
   gotoIfOperation: s32'loop', (IS_LESSER_OR_EQUAL), (VARIABLE), (LoopCount), (STATIC), 120
   endState: 
 
 beginState: s32'finalflash_shadow'
   upon: (IMMEDIATE)
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     jumpToStateInObject: (SELF), s32'end'
     endUpon: 
   sprite: s32'null', 2147483647
   linkParticle: s32'cmn_shadowLine_loop'
   endLabel: 
   beginLabel: s32'end'
   sprite: s32'null', 1
   endLabel: 
   endState: 
 
 beginState: s32'404Eff'
   upon: (IMMEDIATE)
     linkParticle: s32'vgs_401Eff00'
     LinkObjectPosition: 2
     LinkObjectStop: 2
     LinkObjectChangeToEliminate: 2
     jumpToStateInObject: (SELF), s32'End'
     endUpon: 
   sprite: s32'null', 2147483647
   createParticle: s32'vgs_401EffNokosi', (PLAYER)
   endLabel: 
   beginLabel: s32'End'
   sprite: s32'null', 15
   linkParticle: s32'vgs_401Eff01'
   LinkObjectStop: 0
   endLabel: 
   endState: 
 
 beginState: s32'404Eff2'
   upon: (IMMEDIATE)
     linkParticle: s32'vgs_401Eff02'
     LinkObjectPosition: 2
     LinkObjectStop: 2
     LinkObjectChangeToEliminate: 2
     jumpToStateInObject: (SELF), s32'End'
     endUpon: 
   sprite: s32'null', 2147483647
   endLabel: 
   beginLabel: s32'End'
   sprite: s32'null', 15
   linkParticle: s32'vgs_401Eff03'
   LinkObjectStop: 0
   endLabel: 
   endState: 
 
 beginState: s32'FinalFlash_aura00'
   upon: (IMMEDIATE)
     IgnoreFinishStop: 1
     IgnoreWorldStopByAll: 1
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     linkParticle: s32'vgs_432_bodyaura00'
     modifyPositionY: 250000
     setObjectSize: 1200
     jumpToStateInObject: (SELF), s32'End'
     endUpon: 
   sprite: s32'null', 2147483647
   beginLabel: s32'End'
   sprite: s32'null', 10
   endLabel: 
   endState: 
 
 beginState: s32'FF_groundSmokeMatome'
   upon: (IMMEDIATE)
     LinkObjectChangeToEliminate: 2
     IgnoreWorldStopByAll: 1
     endUpon: 
   beginLabel: s32'loop'
   sprite: s32'null', 10
   createObject: s32'FF_groundSmoke_R', (PLAYER)
   createObject: s32'FF_groundSmoke_L', (PLAYER)
   gotoLabel: s32'loop'
   endState: 
 
 beginState: s32'FF_groundSmoke_R'
   upon: (IMMEDIATE)
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     SetAngleY: 130000
     setObjectSize: 3000
     modifyPositionX: 150000
     modifyPositionY: -75000
     addDepth: -300000
     IgnoreWorldStopByAll: 1
     endUpon: 
   sprite: s32'null', 40
   LinkParticleArg: s32'bg_undersmoke00'
   endLabel: 
   endState: 
 
 beginState: s32'FF_groundSmoke_L'
   upon: (IMMEDIATE)
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     SetAngleY: -130000
     setObjectSize: 3000
     modifyPositionX: 150000
     modifyPositionY: -75000
     addDepth: 300000
     IgnoreWorldStopByAll: 1
     endUpon: 
   sprite: s32'null', 40
   LinkParticleArg: s32'bg_undersmoke00'
   endLabel: 
   endState: 
 
 beginState: s32'FF_SyutyuLine'
   upon: (IMMEDIATE)
     LinkObjectChangeToEliminate: 2
     IgnoreWorldStopByAll: 1
     linkParticle: s32'cmn_syutyusen'
     endUpon: 
   sprite: s32'null', 2147483647
   endState: 
 
 beginState: s32'FF_tameShere'
   upon: (IMMEDIATE)
     IgnoreFinishStop: 1
     IgnoreWorldStopByAll: 1
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     linkParticle: s32'vgs_432_tamesphere'
     modifyPositionY: 300000
     modifyPositionX: 250000
     setObjectSize: 350
     endUpon: 
   sprite: s32'null', 2147483647
   endState: 
 
 beginState: s32'FF_Flash'
   upon: (IMMEDIATE)
     LinkObjectChangeToEliminate: 2
     IgnoreWorldStopByAll: 1
     linkParticle: s32'vgs_432_flash00'
     endUpon: 
   sprite: s32'null', 2147483647
   endState: 
 
 beginState: s32'FF_groundSmokeMatome2'
   upon: (IMMEDIATE)
     LinkObjectChangeToEliminate: 2
     IgnoreWorldStopByAll: 1
     setPositionY: 0
     modifyPositionX: 300000
     endUpon: 
   beginLabel: s32'loop'
   sprite: s32'null', 10
   createObject: s32'FF_groundSmoke2', (PLAYER)
   gotoLabel: s32'loop'
   endState: 
 
 beginState: s32'FF_groundSmoke2'
   upon: (IMMEDIATE)
     LinkObjectPosition: 2
     SetAngleY: 185000
     setObjectSize: 1600
     addDepth: 350000
     IgnoreWorldStopByAll: 1
     IgnoreFinishStop: 1
     endUpon: 
   sprite: s32'null', 30
   LinkParticleArg: s32'bg_undersmoke00'
   endLabel: 
   endState: 
 
 beginState: s32'vgs600_scrFlash'
   upon: (IMMEDIATE)
     IgnoreWorldStopByAll: 1
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     linkParticle: s32'vgs_600scranim01'
     flipDirection: 
     endUpon: 
   sprite: s32'null', 2147483647
   endState: 
 
 beginState: s32'vgs600_scranim00'
   upon: (IMMEDIATE)
     IgnoreWorldStopByAll: 1
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     linkParticle: s32'vgs_600scranim00'
     flipDirection: 
     endUpon: 
   sprite: s32'null', 79
   endState: 
 
 beginState: s32'vgs600_BodyAura'
   upon: (IMMEDIATE)
     IgnoreWorldStopByAll: 1
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     SetAngleY: -450000
     LinkParticleArg: s32'cmn_Saura_loop'
     jumpToStateInObject: (SELF), s32'auramove'
     endUpon: 
   sprite: s32'null', 4
   setObjectSize: 500
   sprite: s32'null', 4
   setObjectSize: 600
   sprite: s32'null', 4
   setObjectSize: 650
   sprite: s32'null', 4
   setObjectSize: 700
   sprite: s32'null', 4
   setObjectSize: 750
   beginLabel: s32'loop'
   sprite: s32'null', 3
   setObjectSize: 800
   sprite: s32'null', 3
   setObjectSize: 850
   gotoLabel: s32'loop'
   beginLabel: s32'auramove'
   sprite: s32'null', 2147483647
   modifyPositionX: -100000
   addDepth: -100000
   addObjectHeight: 200
   endState: 
 
 beginState: s32'vgs600_AuraPlus'
   upon: (IMMEDIATE)
     IgnoreWorldStopByAll: 1
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     LinkParticleArg: s32'vgs_600auraplus'
     endUpon: 
   sprite: s32'null', 2147483647
   endState: 
 
 beginState: s32'vgs600_Speed'
   upon: (IMMEDIATE)
     LinkObjectChangeToEliminate: 2
     LinkObjectPosition: 2
     LinkParticleArg: s32'vgs_600speed00'
     SetAngleY: -450000
     endUpon: 
   sprite: s32'null', 24
   endState: 
 
 beginState: s32'vgs600_groundsmoke'
   upon: (IMMEDIATE)
     LinkObjectStop: 2
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     modifyPositionX: 150000
     addDepth: -75000
     endUpon: 
   beginLabel: s32'loop'
   sprite: s32'null', 10
   CreateArg_Scale: 0, 1400
   CreateArg_AngleY: 0, 90000
   CreateParticleArg: s32'bg_undersmoke00', 100
   gotoLabel: s32'loop'
   endState: 
 
 beginState: s32'vgs600_groundsmoke2'
   upon: (IMMEDIATE)
     LinkObjectStop: 2
     LinkObjectPosition: 2
     LinkObjectChangeToEliminate: 2
     modifyPositionX: 150000
     addDepth: 75000
     endUpon: 
   beginLabel: s32'loop'
   sprite: s32'null', 10
   CreateArg_Scale: 0, 1400
   CreateArg_AngleY: 0, -90000
   CreateParticleArg: s32'bg_undersmoke00', 100
   gotoLabel: s32'loop'
   endState: 
 
 beginState: s32'FinishCamera_NmlAtk2C'
   upon: (IMMEDIATE)
     IgnoreWorldStopByFriend: 1
     IgnoreFinishStop: 1
     resetCamera: 
     LinkCommonPawn: 
     endUpon: 
   sprite: s32'null', 60
   createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
   cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
   cameraShake: s32'ShakeTate', 1500, 3, 25, 3
   cameraMotion: -110297, 122176, -13524, 7099, 8611, 0, 72, 0, 90, 30, 1
   sprite: s32'null', 30
   deactivateObjectByName: s32'cmn_syutyusen'
   endState: 
 
 beginState: s32'FinishCamera_NmlAtkAir2C'
   upon: (IMMEDIATE)
     IgnoreWorldStopByFriend: 1
     IgnoreFinishStop: 1
     resetCamera: 
     LinkCommonPawn: 
     endUpon: 
   sprite: s32'null', 60
   createObject: s32'cmn_syutyusen', (ENEMY_CENTER)
   cmnSfx: s32'ARC_BTL_CMN_Hit_XLarge'
   cameraShake: s32'ShakeTate', 1500, 3, 25, 3
   cameraMotion: -54651, 151288, -14477, 7534, 5257, 0, 72, 0, 90, 30, 1
   sprite: s32'null', 30
   deactivateObjectByName: s32'cmn_syutyusen'
   endState: 
 
 beginState: s32'DUMMYDUMMY'
   sprite: s32'null', 1
   endState: 
 
